<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Workflow for interactive visualization of SummarizedExperiment objects with iSEE • iSEEWorkshop</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Workflow for interactive visualization of SummarizedExperiment objects with iSEE">
<meta property="og:description" content="">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">iSEEWorkshop</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/iSEE-lab.html">Workflow for interactive visualization of SummarizedExperiment objects with iSEE</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><link href="iSEE-lab_files/font-awesome-5.1.0/css/all.css" rel="stylesheet">
<link href="iSEE-lab_files/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet">
<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Workflow for interactive visualization of SummarizedExperiment objects with iSEE</h1>
                        <h4 class="author">Kevin Rue-Albrecht, Charlotte Soneson, Federico Marini, and Aaron Lun</h4>
            
            <h4 class="date">May 13, 2019</h4>
      
      
      <div class="hidden name"><code>iSEE-lab.Rmd</code></div>

    </div>

    
    
<div id="instructor-names-and-contact-information" class="section level1">
<h1 class="hasAnchor">
<a href="#instructor-names-and-contact-information" class="anchor"></a>Instructor names and contact information</h1>
<ul>
<li>Kevin Rue-Albrecht<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a> (<a href="mailto:kevin.rue-albrecht@kennedy.ox.ac.uk">kevin.rue-albrecht@kennedy.ox.ac.uk</a>)</li>
<li>Charlotte Soneson<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a> (<a href="mailto:charlotte.soneson@fmi.ch">charlotte.soneson@fmi.ch</a>)</li>
</ul>
</div>
<div id="workshop-description" class="section level1">
<h1 class="hasAnchor">
<a href="#workshop-description" class="anchor"></a>Workshop description</h1>
<p>This workshop demonstrates how to use the <em><a href="https://bioconductor.org/packages/3.10/iSEE">iSEE</a></em> package to create and configure interactive applications for the exploration of various types of genomics data sets (e.g., bulk and single-cell RNA-seq, CyTOF, gene expression microarray).</p>
<p>This workshop will be presented as a lab session that combines an instructor-led live demo, followed by hands-on experimentation guided by completely worked examples and stand-alone notes that participants may continue to use after the workshop.</p>
<p>The instructor-led live demo comprises three parts:</p>
<ol style="list-style-type: decimal">
<li>Brief lecture on the package concept and functionality</li>
<li>Overview of the graphical user interface</li>
<li>Instructions to preconfigure <em><a href="https://bioconductor.org/packages/3.10/iSEE">iSEE</a></em> apps</li>
</ol>
<p>The hands-on lab comprises three parts:</p>
<ol style="list-style-type: decimal">
<li>Inspection of single-cell RNA-seq data at various steps of a typical computational workflow, including quality control and dimensionality reduction</li>
<li>Addition of custom panels to the user interface for advanced visualization.</li>
<li>Additional questions from the participants, including individual use cases and suggestions for future developments</li>
</ol>
<p>Participants are encouraged to ask questions at any time during the workshop.</p>
<div id="prerequisites" class="section level2">
<h2 class="hasAnchor">
<a href="#prerequisites" class="anchor"></a>Pre-requisites</h2>
<ul>
<li>Basic knowledge of <a href="https://www.r-project.org/about.html"><em>R</em></a> syntax and the use of data-frames</li>
<li>Familiarity with the <em><a href="https://bioconductor.org/packages/3.10/SummarizedExperiment">SummarizedExperiment</a></em> and <em><a href="https://bioconductor.org/packages/3.10/SingleCellExperiment">SingleCellExperiment</a></em> classes</li>
<li>Familiarity with the <em><a href="https://CRAN.R-project.org/package=shiny">shiny</a></em> CRAN package</li>
<li>Familiarity with the <em><a href="https://bioconductor.org/packages/3.10/TENxPBMCData">TENxPBMCData</a></em> package and vignette</li>
</ul>
<p>Additional background reading about the programming environment, relevant packages, and example use cases:</p>
<ul>
<li>Shiny from RStudio: <a href="https://shiny.rstudio.com" class="uri">https://shiny.rstudio.com</a>
</li>
<li>SummarizedExperiment paper: <a href="https://www.nature.com/articles/nmeth.3252" class="uri">https://www.nature.com/articles/nmeth.3252</a> (Figure 2)</li>
<li>iSEE paper: <a href="https://f1000research.com/articles/7-741/v1" class="uri">https://f1000research.com/articles/7-741/v1</a>
</li>
</ul>
</div>
<div id="participation" class="section level2">
<h2 class="hasAnchor">
<a href="#participation" class="anchor"></a>Workshop Participation</h2>
<p>Students will participate by following along an <a href="https://rmarkdown.rstudio.com/"><em>R markdown</em></a> document, and asking questions throughout the workshop. There is also scope for participants to apply <em><a href="https://bioconductor.org/packages/3.10/iSEE">iSEE</a></em> to their own data sets, and fuel the discussion with more questions about specific use cases.</p>
</div>
<div id="packages" class="section level2">
<h2 class="hasAnchor">
<a href="#packages" class="anchor"></a>R/Bioconductor packages used</h2>
<ol style="list-style-type: decimal">
<li>
<em>iSEE</em> : <a href="https://bioconductor.org/packages/iSEE" class="uri">https://bioconductor.org/packages/iSEE</a>
</li>
<li>
<em>SummarizedExperiment</em>: <a href="https://bioconductor.org/packages/SummarizedExperiment" class="uri">https://bioconductor.org/packages/SummarizedExperiment</a>
</li>
<li>
<em>SingleCellExperiment</em>: <a href="https://bioconductor.org/packages/SingleCellExperiment" class="uri">https://bioconductor.org/packages/SingleCellExperiment</a>
</li>
<li>
<em>TENxPBMCData</em>: <a href="https://bioconductor.org/packages/TENxPBMCData" class="uri">https://bioconductor.org/packages/TENxPBMCData</a>
</li>
<li>
<em>scater</em>: <a href="https://bioconductor.org/packages/scater" class="uri">https://bioconductor.org/packages/scater</a>
</li>
<li>
<em>scran</em>: <a href="https://bioconductor.org/packages/scran" class="uri">https://bioconductor.org/packages/scran</a>
</li>
<li>
<em>BiocSingular</em>: <a href="https://bioconductor.org/packages/BiocSingular" class="uri">https://bioconductor.org/packages/BiocSingular</a>
</li>
</ol>
</div>
<div id="outline" class="section level2">
<h2 class="hasAnchor">
<a href="#outline" class="anchor"></a>Time outline</h2>
<table class="table">
<thead><tr class="header">
<th>Activity</th>
<th>Time</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>
<strong>Lecture</strong>: Overview of package and concepts</td>
<td>15m</td>
</tr>
<tr class="even">
<td>
<strong>Live demo</strong>: the user interface</td>
<td>15m</td>
</tr>
<tr class="odd">
<td>
<strong>Lab</strong>: A single-cell RNA-seq workflow</td>
<td>20m</td>
</tr>
<tr class="even">
<td>
<strong>Lab</strong>: Configuring the app interface</td>
<td>20m</td>
</tr>
<tr class="odd">
<td>
<strong>Lab</strong>: Custom panels</td>
<td>20m</td>
</tr>
<tr class="even">
<td>Additional questions</td>
<td>15m</td>
</tr>
</tbody>
</table>
<p><strong>Total:</strong> 1h45</p>
</div>
</div>
<div id="goals-objectives" class="section level1">
<h1 class="hasAnchor">
<a href="#goals-objectives" class="anchor"></a>Workshop goals and objectives</h1>
<div id="goals" class="section level2">
<h2 class="hasAnchor">
<a href="#goals" class="anchor"></a>Learning goals</h2>
<ul>
<li>Recognize the benefits of integrative data containers such as <a href="http://bioconductor.org/packages/SummarizedExperiment"><code>SummarizedExperiment</code></a> and <a href="http://bioconductor.org/packages/SingleCellExperiment"><code>SingleCellExperiment</code></a> for downstream analyses and visualization</li>
<li>Outline the unique features of <a href="http://bioconductor.org/packages/iSEE/"><em>iSEE</em></a> built upon the <a href="https://shiny.rstudio.com">RStudio Shiny</a> framework</li>
<li>Identify aspcts of biological data that may be combined into insightful graphical outputs</li>
<li>Utilize interactive GUI components and layouts to efficiently extract information from biological data sets</li>
<li>Describe how to construct interactive apps and custom panels</li>
</ul>
</div>
<div id="objectives" class="section level2">
<h2 class="hasAnchor">
<a href="#objectives" class="anchor"></a>Learning objectives</h2>
<ul>
<li>Memorize the key information available in <a href="http://bioconductor.org/packages/SummarizedExperiment"><code>SummarizedExperiment</code></a> and <a href="http://bioconductor.org/packages/SingleCellExperiment"><code>SingleCellExperiment</code></a> objects</li>
<li>Set up a local environment for running <a href="https://bioconductor.org/packages/iSEE"><em>iSEE</em></a> apps</li>
<li>Interact with components of the <a href="https://bioconductor.org/packages/iSEE"><em>iSEE</em></a> user interface to visually inspect and discuss various data sets</li>
<li>Identify and locate configurable aspects of <a href="https://bioconductor.org/packages/iSEE"><em>iSEE</em></a> apps</li>
<li>Practice interactive visualization over a single-cell RNA-sequencing workflow</li>
<li>Design custom <a href="https://bioconductor.org/packages/iSEE"><em>iSEE</em></a> panels for advanced use cases</li>
<li>Imagine use cases and future developments for interactive visualization as part of computational workflows</li>
</ul>
</div>
</div>
<div id="prepare-scrnaseq-data" class="section level1">
<h1 class="hasAnchor">
<a href="#prepare-scrnaseq-data" class="anchor"></a>Preparation of example scRNAseq data</h1>
<p>In this workshop, we use example data from the <em><a href="https://bioconductor.org/packages/3.10/TENxPBMCData">TENxPBMCData</a></em> package. This package provides an <em>R / Bioconductor</em> resource for representing and manipulating different single-cell RNA-seq data sets profiling peripheral blood mononuclear cells (PBMC) generated by <a href="https://support.10xgenomics.com/single-cell-gene-expression/datasets">10X Genomics</a>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(TENxPBMCData)</a></code></pre></div>
<p>The man page for the <code><a href="https://www.rdocumentation.org/packages/TENxPBMCData/topics/TENxPBMCData">TENxPBMCData()</a></code> function gives an idea of the datasets that are available from this package. It can be opened with the following command.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/help">help</a></span>(TENxPBMCData)</a></code></pre></div>
<p>Here, we use the <code>"pbmc3k"</code> dataset, which contains gene expression profiles for 2,700 single peripheral blood mononuclear cells. The first time this dataset is loaded, this command downloads the dataset to a local cache, which takes some time, depending on the speed of your internet connection. Subsequent times, the same command loads the dataset directly from the local cache.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">sce &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/TENxPBMCData/topics/TENxPBMCData">TENxPBMCData</a></span>(<span class="dt">dataset =</span> <span class="st">"pbmc3k"</span>)</a></code></pre></div>
<p>At this point we can inspect the dataset in the console.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">sce</a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="co">#&gt; class: SingleCellExperiment </span></a>
<a class="sourceLine" id="cb4-3" data-line-number="3"><span class="co">#&gt; dim: 32738 2700 </span></a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="co">#&gt; metadata(0):</span></a>
<a class="sourceLine" id="cb4-5" data-line-number="5"><span class="co">#&gt; assays(1): counts</span></a>
<a class="sourceLine" id="cb4-6" data-line-number="6"><span class="co">#&gt; rownames(32738): ENSG00000243485 ENSG00000237613 ...</span></a>
<a class="sourceLine" id="cb4-7" data-line-number="7"><span class="co">#&gt;   ENSG00000215616 ENSG00000215611</span></a>
<a class="sourceLine" id="cb4-8" data-line-number="8"><span class="co">#&gt; rowData names(3): ENSEMBL_ID Symbol_TENx Symbol</span></a>
<a class="sourceLine" id="cb4-9" data-line-number="9"><span class="co">#&gt; colnames: NULL</span></a>
<a class="sourceLine" id="cb4-10" data-line-number="10"><span class="co">#&gt; colData names(11): Sample Barcode ... Individual Date_published</span></a>
<a class="sourceLine" id="cb4-11" data-line-number="11"><span class="co">#&gt; reducedDimNames(0):</span></a>
<a class="sourceLine" id="cb4-12" data-line-number="12"><span class="co">#&gt; spikeNames(0):</span></a></code></pre></div>
<p>The dataset is provided as an object of the <code>SingleCellExperiment</code> class. In particular, this summary view indicates that the following pieces of information are available:</p>
<ul>
<li>An assay matrix named <code>"counts"</code>
</li>
<li>Row names (i.e. genes) are Ensembl gene IDs</li>
<li>Row metadata include for each gene the official gene symbol, and the gene symbol used by the 10x CellRanger quantification pipeline</li>
<li>Column names (i.e., cells) are not initialized and left to <code>NULL</code>
</li>
<li>Column metadata include diverse information for each cell, including the cell barcode (<code>"Barcode"</code>) and the donor identifier (<code>"Individual"</code>).</li>
</ul>
<p>Note that a <code>SingleCellExperiment</code> object (or, more generally, any <code>SummarizedExperiment</code> object) like this one already contains sufficient information to launch an interactive application instance to visualize the available data and metadata, using the <code><a href="https://www.rdocumentation.org/packages/iSEE/topics/iSEE">iSEE()</a></code> function.</p>
<p>For the purpose of this workshop, we first apply some preprocessing to the <code>SingleCellExperiment</code> object, in order to populate it with more information that can be visualized with <code>iSEE</code>. The steps below roughly follow those outlined in the <a href="https://bioconductor.org/packages/simpleSingleCell"><em>simpleSingleCell</em></a> Bioconductor workflow.</p>
<p>We start by adding column names to the object, and use gene symbols instead of Ensembl IDs as row names. In the case where multiple Ensembl identifiers correspond to the same gene symbol, the <code><a href="https://www.rdocumentation.org/packages/scater/topics/uniquifyFeatureNames">scater::uniquifyFeatureNames</a></code> function concatenates the Ensembl ID and the gene symbol in order to generate unique feature names.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(scater)</a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(sce) &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste0</a></span>(<span class="st">"Cell"</span>, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/seq">seq_len</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/nrow">ncol</a></span>(sce)))</a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(sce) &lt;-<span class="st"> </span>scater<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/scater/topics/uniquifyFeatureNames">uniquifyFeatureNames</a></span>(</a>
<a class="sourceLine" id="cb5-4" data-line-number="4">    <span class="dt">ID =</span> <span class="kw">rowData</span>(sce)<span class="op">$</span>ENSEMBL_ID,</a>
<a class="sourceLine" id="cb5-5" data-line-number="5">    <span class="dt">names =</span> <span class="kw">rowData</span>(sce)<span class="op">$</span>Symbol_TENx</a>
<a class="sourceLine" id="cb5-6" data-line-number="6">)</a>
<a class="sourceLine" id="cb5-7" data-line-number="7"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/head">head</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(sce))</a>
<a class="sourceLine" id="cb5-8" data-line-number="8"><span class="co">#&gt; [1] "MIR1302-10"   "FAM138A"      "OR4F5"        "RP11-34P13.7"</span></a>
<a class="sourceLine" id="cb5-9" data-line-number="9"><span class="co">#&gt; [5] "RP11-34P13.8" "AL627309.1"</span></a></code></pre></div>
<p>Next, we use the <em><a href="https://bioconductor.org/packages/3.10/scater">scater</a></em> package to calculate gene- and cell-level quality metrics. These metrics are added as columns to the <code>rowData</code> and <code>colData</code> slots of the <code>SingleCellExperiment</code> object, respectively.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">MT &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(sce)[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/grep">grep</a></span>(<span class="st">"^MT-"</span>, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(sce))]</a>
<a class="sourceLine" id="cb6-2" data-line-number="2">sce &lt;-<span class="st"> </span>scater<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/scater/topics/calculateQCMetrics">calculateQCMetrics</a></span>(<span class="dt">object =</span> sce, </a>
<a class="sourceLine" id="cb6-3" data-line-number="3">                                  <span class="dt">feature_controls =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(<span class="dt">MT =</span> MT))</a>
<a class="sourceLine" id="cb6-4" data-line-number="4">sce</a>
<a class="sourceLine" id="cb6-5" data-line-number="5"><span class="co">#&gt; class: SingleCellExperiment </span></a>
<a class="sourceLine" id="cb6-6" data-line-number="6"><span class="co">#&gt; dim: 32738 2700 </span></a>
<a class="sourceLine" id="cb6-7" data-line-number="7"><span class="co">#&gt; metadata(0):</span></a>
<a class="sourceLine" id="cb6-8" data-line-number="8"><span class="co">#&gt; assays(1): counts</span></a>
<a class="sourceLine" id="cb6-9" data-line-number="9"><span class="co">#&gt; rownames(32738): MIR1302-10 FAM138A ... AC002321.2 AC002321.1</span></a>
<a class="sourceLine" id="cb6-10" data-line-number="10"><span class="co">#&gt; rowData names(11): ENSEMBL_ID Symbol_TENx ... total_counts</span></a>
<a class="sourceLine" id="cb6-11" data-line-number="11"><span class="co">#&gt;   log10_total_counts</span></a>
<a class="sourceLine" id="cb6-12" data-line-number="12"><span class="co">#&gt; colnames(2700): Cell1 Cell2 ... Cell2699 Cell2700</span></a>
<a class="sourceLine" id="cb6-13" data-line-number="13"><span class="co">#&gt; colData names(47): Sample Barcode ...</span></a>
<a class="sourceLine" id="cb6-14" data-line-number="14"><span class="co">#&gt;   pct_counts_in_top_200_features_MT</span></a>
<a class="sourceLine" id="cb6-15" data-line-number="15"><span class="co">#&gt;   pct_counts_in_top_500_features_MT</span></a>
<a class="sourceLine" id="cb6-16" data-line-number="16"><span class="co">#&gt; reducedDimNames(0):</span></a>
<a class="sourceLine" id="cb6-17" data-line-number="17"><span class="co">#&gt; spikeNames(0):</span></a></code></pre></div>
<p>We filter out a few cells with a large fraction of the counts coming from mitochondrial genes, since these may be damaged cells. Notice the reduced number of columns in the dataset below.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">(sce &lt;-<span class="st"> </span>sce[, sce<span class="op">$</span>pct_counts_MT <span class="op">&lt;</span><span class="st"> </span><span class="dv">5</span>])</a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="co">#&gt; class: SingleCellExperiment </span></a>
<a class="sourceLine" id="cb7-3" data-line-number="3"><span class="co">#&gt; dim: 32738 2643 </span></a>
<a class="sourceLine" id="cb7-4" data-line-number="4"><span class="co">#&gt; metadata(0):</span></a>
<a class="sourceLine" id="cb7-5" data-line-number="5"><span class="co">#&gt; assays(1): counts</span></a>
<a class="sourceLine" id="cb7-6" data-line-number="6"><span class="co">#&gt; rownames(32738): MIR1302-10 FAM138A ... AC002321.2 AC002321.1</span></a>
<a class="sourceLine" id="cb7-7" data-line-number="7"><span class="co">#&gt; rowData names(11): ENSEMBL_ID Symbol_TENx ... total_counts</span></a>
<a class="sourceLine" id="cb7-8" data-line-number="8"><span class="co">#&gt;   log10_total_counts</span></a>
<a class="sourceLine" id="cb7-9" data-line-number="9"><span class="co">#&gt; colnames(2643): Cell1 Cell2 ... Cell2699 Cell2700</span></a>
<a class="sourceLine" id="cb7-10" data-line-number="10"><span class="co">#&gt; colData names(47): Sample Barcode ...</span></a>
<a class="sourceLine" id="cb7-11" data-line-number="11"><span class="co">#&gt;   pct_counts_in_top_200_features_MT</span></a>
<a class="sourceLine" id="cb7-12" data-line-number="12"><span class="co">#&gt;   pct_counts_in_top_500_features_MT</span></a>
<a class="sourceLine" id="cb7-13" data-line-number="13"><span class="co">#&gt; reducedDimNames(0):</span></a>
<a class="sourceLine" id="cb7-14" data-line-number="14"><span class="co">#&gt; spikeNames(0):</span></a></code></pre></div>
<p>Next, we calculate size factors and normalized and log-transformed expression values, using the <em><a href="https://bioconductor.org/packages/3.10/scran">scran</a></em> and <em><a href="https://bioconductor.org/packages/3.10/scater">scater</a></em> packages.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(scran)</a>
<a class="sourceLine" id="cb8-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(BiocSingular)</a>
<a class="sourceLine" id="cb8-3" data-line-number="3">sce &lt;-<span class="st"> </span>scran<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/scran/topics/computeSumFactors">computeSumFactors</a></span>(sce, <span class="dt">min.mean =</span> <span class="fl">0.1</span>)</a>
<a class="sourceLine" id="cb8-4" data-line-number="4"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/summary">summary</a></span>(<span class="kw">sizeFactors</span>(sce))</a>
<a class="sourceLine" id="cb8-5" data-line-number="5"><span class="co">#&gt;    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span></a>
<a class="sourceLine" id="cb8-6" data-line-number="6"><span class="co">#&gt;  0.1164  0.7309  0.9004  1.0000  1.1635  9.8580</span></a>
<a class="sourceLine" id="cb8-7" data-line-number="7">sce &lt;-<span class="st"> </span>scater<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/scater/topics/normalize">normalize</a></span>(sce)</a></code></pre></div>
<p>In order to extract the most informative genes, we first model the mean-variance trend and decompose the variance into biological and technical components.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw">logcounts</span>(sce) &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/matrix">as.matrix</a></span>(<span class="kw">logcounts</span>(sce))</a>
<a class="sourceLine" id="cb9-2" data-line-number="2">new.trend &lt;-<span class="st"> </span>scran<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/scran/topics/makeTechTrend">makeTechTrend</a></span>(<span class="dt">x =</span> sce)</a>
<a class="sourceLine" id="cb9-3" data-line-number="3">fit &lt;-<span class="st"> </span>scran<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/scran/topics/trendVar">trendVar</a></span>(sce, <span class="dt">use.spikes =</span> <span class="ot">FALSE</span>, <span class="dt">loess.args =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(<span class="dt">span =</span> <span class="fl">0.05</span>))</a>
<a class="sourceLine" id="cb9-4" data-line-number="4">fit<span class="op">$</span>trend &lt;-<span class="st"> </span>new.trend</a>
<a class="sourceLine" id="cb9-5" data-line-number="5">dec &lt;-<span class="st"> </span>scran<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/scran/topics/decomposeVar">decomposeVar</a></span>(<span class="dt">fit =</span> fit)</a>
<a class="sourceLine" id="cb9-6" data-line-number="6">top.dec &lt;-<span class="st"> </span>dec[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/order">order</a></span>(dec<span class="op">$</span>bio, <span class="dt">decreasing =</span> <span class="ot">TRUE</span>), ] </a>
<a class="sourceLine" id="cb9-7" data-line-number="7"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/head">head</a></span>(top.dec)</a>
<a class="sourceLine" id="cb9-8" data-line-number="8"><span class="co">#&gt; DataFrame with 6 rows and 6 columns</span></a>
<a class="sourceLine" id="cb9-9" data-line-number="9"><span class="co">#&gt;                     mean            total              bio</span></a>
<a class="sourceLine" id="cb9-10" data-line-number="10"><span class="co">#&gt;                &lt;numeric&gt;        &lt;numeric&gt;        &lt;numeric&gt;</span></a>
<a class="sourceLine" id="cb9-11" data-line-number="11"><span class="co">#&gt; LYZ     1.63949764898989 3.92208536906663 3.27404036212034</span></a>
<a class="sourceLine" id="cb9-12" data-line-number="12"><span class="co">#&gt; S100A9  1.07322748956473 3.36899492417173 2.73849817385183</span></a>
<a class="sourceLine" id="cb9-13" data-line-number="13"><span class="co">#&gt; FTL     3.67845532888203 2.88498930976467   2.662250196795</span></a>
<a class="sourceLine" id="cb9-14" data-line-number="14"><span class="co">#&gt; HLA-DRA 1.56936865352987 3.16577098114059 2.51202239465566</span></a>
<a class="sourceLine" id="cb9-15" data-line-number="15"><span class="co">#&gt; CD74    2.23652914180461 2.75839122768224 2.21465618657833</span></a>
<a class="sourceLine" id="cb9-16" data-line-number="16"><span class="co">#&gt; FTH1    3.52182996359555 2.20828385314013 1.95849254164614</span></a>
<a class="sourceLine" id="cb9-17" data-line-number="17"><span class="co">#&gt;                      tech   p.value       FDR</span></a>
<a class="sourceLine" id="cb9-18" data-line-number="18"><span class="co">#&gt;                 &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></a>
<a class="sourceLine" id="cb9-19" data-line-number="19"><span class="co">#&gt; LYZ     0.648045006946287         0         0</span></a>
<a class="sourceLine" id="cb9-20" data-line-number="20"><span class="co">#&gt; S100A9  0.630496750319901         0         0</span></a>
<a class="sourceLine" id="cb9-21" data-line-number="21"><span class="co">#&gt; FTL      0.22273911296967         0         0</span></a>
<a class="sourceLine" id="cb9-22" data-line-number="22"><span class="co">#&gt; HLA-DRA 0.653748586484937         0         0</span></a>
<a class="sourceLine" id="cb9-23" data-line-number="23"><span class="co">#&gt; CD74    0.543735041103904         0         0</span></a>
<a class="sourceLine" id="cb9-24" data-line-number="24"><span class="co">#&gt; FTH1    0.249791311493993         0         0</span></a></code></pre></div>
<p>Next, we apply Principal Components Analysis (PCA) and t-distributed Stochastic Neighbor Embedding (t-SNE) to generate low-dimensional representations of the cells in our data set. These low-dimensional representations are added to the <code>reducedDim</code> slot of the <code>SingleCellExperiment</code> object.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Random">set.seed</a></span>(<span class="dv">1000</span>)</a>
<a class="sourceLine" id="cb10-2" data-line-number="2">sce &lt;-<span class="st"> </span>scran<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/scran/topics/denoisePCA">denoisePCA</a></span>(sce, <span class="dt">technical =</span> new.trend, <span class="dt">BSPARAM =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/BiocSingular/topics/BiocSingularParam">IrlbaParam</a></span>())</a>
<a class="sourceLine" id="cb10-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/nrow">ncol</a></span>(<span class="kw">reducedDim</span>(sce, <span class="st">"PCA"</span>))</a>
<a class="sourceLine" id="cb10-4" data-line-number="4"><span class="co">#&gt; [1] 32</span></a>
<a class="sourceLine" id="cb10-5" data-line-number="5"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Random">set.seed</a></span>(<span class="dv">1000</span>)</a>
<a class="sourceLine" id="cb10-6" data-line-number="6">sce &lt;-<span class="st"> </span>scater<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/scater/topics/runTSNE">runTSNE</a></span>(sce, <span class="dt">use_dimred =</span> <span class="st">"PCA"</span>, <span class="dt">perplexity =</span> <span class="dv">30</span>)</a>
<a class="sourceLine" id="cb10-7" data-line-number="7">sce</a>
<a class="sourceLine" id="cb10-8" data-line-number="8"><span class="co">#&gt; class: SingleCellExperiment </span></a>
<a class="sourceLine" id="cb10-9" data-line-number="9"><span class="co">#&gt; dim: 32738 2643 </span></a>
<a class="sourceLine" id="cb10-10" data-line-number="10"><span class="co">#&gt; metadata(1): log.exprs.offset</span></a>
<a class="sourceLine" id="cb10-11" data-line-number="11"><span class="co">#&gt; assays(2): counts logcounts</span></a>
<a class="sourceLine" id="cb10-12" data-line-number="12"><span class="co">#&gt; rownames(32738): MIR1302-10 FAM138A ... AC002321.2 AC002321.1</span></a>
<a class="sourceLine" id="cb10-13" data-line-number="13"><span class="co">#&gt; rowData names(11): ENSEMBL_ID Symbol_TENx ... total_counts</span></a>
<a class="sourceLine" id="cb10-14" data-line-number="14"><span class="co">#&gt;   log10_total_counts</span></a>
<a class="sourceLine" id="cb10-15" data-line-number="15"><span class="co">#&gt; colnames(2643): Cell1 Cell2 ... Cell2699 Cell2700</span></a>
<a class="sourceLine" id="cb10-16" data-line-number="16"><span class="co">#&gt; colData names(47): Sample Barcode ...</span></a>
<a class="sourceLine" id="cb10-17" data-line-number="17"><span class="co">#&gt;   pct_counts_in_top_200_features_MT</span></a>
<a class="sourceLine" id="cb10-18" data-line-number="18"><span class="co">#&gt;   pct_counts_in_top_500_features_MT</span></a>
<a class="sourceLine" id="cb10-19" data-line-number="19"><span class="co">#&gt; reducedDimNames(2): PCA TSNE</span></a>
<a class="sourceLine" id="cb10-20" data-line-number="20"><span class="co">#&gt; spikeNames(0):</span></a></code></pre></div>
<p>Finally, we cluster the cells using a graph-based algorithm, and find ‘marker genes’ for each cluster as the genes that are significantly upregulated in the cluster compared to each of the other inferred clusters. The adjusted p-values from this test, for each cluster, are added to the <code>rowData</code> slot of the object.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">snn.gr &lt;-<span class="st"> </span>scran<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/scran/topics/buildSNNGraph">buildSNNGraph</a></span>(sce, <span class="dt">use.dimred =</span> <span class="st">"PCA"</span>)</a>
<a class="sourceLine" id="cb11-2" data-line-number="2">clusters &lt;-<span class="st"> </span>igraph<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/igraph/topics/cluster_walktrap">cluster_walktrap</a></span>(snn.gr)</a>
<a class="sourceLine" id="cb11-3" data-line-number="3">sce<span class="op">$</span>Cluster &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/factor">factor</a></span>(clusters<span class="op">$</span>membership)</a>
<a class="sourceLine" id="cb11-4" data-line-number="4"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/table">table</a></span>(sce<span class="op">$</span>Cluster)</a>
<a class="sourceLine" id="cb11-5" data-line-number="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb11-6" data-line-number="6"><span class="co">#&gt;   1   2   3   4   5   6   7   8   9  10  11  12 </span></a>
<a class="sourceLine" id="cb11-7" data-line-number="7"><span class="co">#&gt; 197 692 161  31 150 331 162 339 137 422  10  11</span></a>
<a class="sourceLine" id="cb11-8" data-line-number="8"></a>
<a class="sourceLine" id="cb11-9" data-line-number="9">markers &lt;-<span class="st"> </span>scran<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/scran/topics/findMarkers">findMarkers</a></span>(sce, <span class="dt">clusters =</span> sce<span class="op">$</span>Cluster, </a>
<a class="sourceLine" id="cb11-10" data-line-number="10">                              <span class="dt">direction =</span> <span class="st">"up"</span>, <span class="dt">pval.type =</span> <span class="st">"all"</span>)</a>
<a class="sourceLine" id="cb11-11" data-line-number="11"><span class="cf">for</span> (i <span class="cf">in</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(markers)) {</a>
<a class="sourceLine" id="cb11-12" data-line-number="12">    <span class="kw">rowData</span>(sce)[, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste0</a></span>(<span class="st">"FDR_cluster"</span>, i)] &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb11-13" data-line-number="13"><span class="st">        </span>markers[[i]]<span class="op">$</span>FDR[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/match">match</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(sce), </a>
<a class="sourceLine" id="cb11-14" data-line-number="14">                               <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(markers[[i]]))]</a>
<a class="sourceLine" id="cb11-15" data-line-number="15">}</a>
<a class="sourceLine" id="cb11-16" data-line-number="16">sce</a>
<a class="sourceLine" id="cb11-17" data-line-number="17"><span class="co">#&gt; class: SingleCellExperiment </span></a>
<a class="sourceLine" id="cb11-18" data-line-number="18"><span class="co">#&gt; dim: 32738 2643 </span></a>
<a class="sourceLine" id="cb11-19" data-line-number="19"><span class="co">#&gt; metadata(1): log.exprs.offset</span></a>
<a class="sourceLine" id="cb11-20" data-line-number="20"><span class="co">#&gt; assays(2): counts logcounts</span></a>
<a class="sourceLine" id="cb11-21" data-line-number="21"><span class="co">#&gt; rownames(32738): MIR1302-10 FAM138A ... AC002321.2 AC002321.1</span></a>
<a class="sourceLine" id="cb11-22" data-line-number="22"><span class="co">#&gt; rowData names(23): ENSEMBL_ID Symbol_TENx ... FDR_cluster11</span></a>
<a class="sourceLine" id="cb11-23" data-line-number="23"><span class="co">#&gt;   FDR_cluster12</span></a>
<a class="sourceLine" id="cb11-24" data-line-number="24"><span class="co">#&gt; colnames(2643): Cell1 Cell2 ... Cell2699 Cell2700</span></a>
<a class="sourceLine" id="cb11-25" data-line-number="25"><span class="co">#&gt; colData names(48): Sample Barcode ...</span></a>
<a class="sourceLine" id="cb11-26" data-line-number="26"><span class="co">#&gt;   pct_counts_in_top_500_features_MT Cluster</span></a>
<a class="sourceLine" id="cb11-27" data-line-number="27"><span class="co">#&gt; reducedDimNames(2): PCA TSNE</span></a>
<a class="sourceLine" id="cb11-28" data-line-number="28"><span class="co">#&gt; spikeNames(0):</span></a></code></pre></div>
<p>This concludes the preparation of the data. We have now a <code>SingleCellExperiment</code> object that contains different types of abundance values, representations in reduced dimensions, as well as a range of row (feature) and column (cell) metadata.</p>
<p>We can launch an <em><a href="https://bioconductor.org/packages/3.10/iSEE">iSEE</a></em> instance for exploring this data set using the <code><a href="https://www.rdocumentation.org/packages/iSEE/topics/iSEE">iSEE()</a></code> function:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(iSEE)</a>
<a class="sourceLine" id="cb12-2" data-line-number="2"><span class="cf">if</span> (<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/interactive">interactive</a></span>()) {</a>
<a class="sourceLine" id="cb12-3" data-line-number="3">    <span class="kw"><a href="https://www.rdocumentation.org/packages/iSEE/topics/iSEE">iSEE</a></span>(sce)</a>
<a class="sourceLine" id="cb12-4" data-line-number="4">}</a></code></pre></div>
<p>The main argument to the <code><a href="https://www.rdocumentation.org/packages/iSEE/topics/iSEE">iSEE()</a></code> function is a <code>SummarizedExperiment</code> object, or an object of any class extending <code>SummarizedExperiment</code> (such as <code>SingleCellExperiment</code>, in this case). No other restrictions are made on the type of data stored in the object, and <em><a href="https://bioconductor.org/packages/3.10/iSEE">iSEE</a></em> can be used for interactive visualization of many different types of data. It is also worth noting that for various types of data, Bioconductor packages provides functionality for directly importing quantifications generated by external software packages into a <code>SummarizedExperiment</code> object. For example, the <em><a href="https://bioconductor.org/packages/3.10/DropletUtils">DropletUtils</a></em> package can read quantifications from the 10x Genomics CellRanger pipeline for single-cell RNA-seq data, and the <em><a href="https://bioconductor.org/packages/3.10/tximeta">tximeta</a></em> package can be used to read data from transcript quantification pipelines into a <code>SummarizedExperiment</code> object.</p>
</div>
<div id="isee-overview" class="section level1">
<h1 class="hasAnchor">
<a href="#isee-overview" class="anchor"></a>Overview of the <em>iSEE</em> package</h1>
<!-- Talk a bit about the usefulness of visualization. -->
<p>This section provides an overview of the graphical interface of <em><a href="https://bioconductor.org/packages/3.10/iSEE">iSEE</a></em> applications. To follow along, make sure that you have launched an <em><a href="https://bioconductor.org/packages/3.10/iSEE">iSEE</a></em> instance as described in the code block at the end of the previous section. Note that in the default configuration, the panels do not look exactly like the ones shown in the screenshots below. For example, data points are not immediately colored, and the default annotation variables displayed by each panel may differ. Later sections of this workshop demonstrate how to modify the content of the panels and how they are displayed.</p>
<p>Note that for simplicity, we typically refer to a <code>SummarizedExperiment</code> in this workshop; however, <em><a href="https://bioconductor.org/packages/3.10/iSEE">iSEE</a></em> works seamlessly for objects of any class extending <code>SummarizedExperiment</code> as well (e.g., <code>SingleCellExperiment</code>, <code>DESeqDataSet</code>). That said, some types of panels – such as the <code>Reduced dimension plot</code> – are only available for objects that contain a <code>reducedDim</code> slot (in particular, <code>SingleCellExperiment</code> objects); the basic <code>SummarizedExperiment</code> class does not contain this slot. In this workshop, we refer to the rows of the <code>SummarizedExperiment</code> object as ‘features’ (these can be genes, transcripts, genomic regions, etc) and to the columns as ‘samples’ (which, in our example data set, are single cells).</p>
<div id="user-interface" class="section level2">
<h2 class="hasAnchor">
<a href="#user-interface" class="anchor"></a>The user interface</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.10/iSEE">iSEE</a></em> user interface consists of a number of panels, each displaying the data provided in the <code>SummarizedExperiment</code> from a specific perspective. There are 8 standard panel types; 6 plot panels and 2 table panels, all showcased in the figure below. In the default configuration, one panel of each type is included when launching the <em><a href="https://bioconductor.org/packages/3.10/iSEE">iSEE</a></em> user interface. However, users are free to rearrange, resize, remove or add panels freely, as described in a <a href="#configuration-ui">later section</a>. We provide a brief overview of each panel type in the following subsections.</p>
<p>In addition to the 8 standard panel types, user can create custom panels (both plots and tables). The creation and configuration of custom panels is discussed later, in a <a href="#custom-panels">dedicated section</a> of this workshop.</p>
<!-- This is perhaps not very pedagogical - the app will not look like the one below if it is just opened with the object created above.
But it's nice to show the capabilities of the panels too, not just the default settings.
I added a few sentences to the introduction of this section to try to address this. -->
<p><img src="img/iSEEinterface.png"></p>
<div id="reddimplot" class="section level3">
<h3 class="hasAnchor">
<a href="#reddimplot" class="anchor"></a>Reduced dimension plot</h3>
<p>The reduced dimension plot can display any reduced dimension representation that is present in the <code>reducedDim</code> slot of the <code>SingleCellExperiment</code> object.</p>
<p>Note that this slot is not defined for the base <code>SummarizedExperiment</code> class, in which case the user interface does not allow the inclusion of panels of this type.</p>
<p><img src="img/redDimPlot.png"></p>
</div>
<div id="coldataplot" class="section level3">
<h3 class="hasAnchor">
<a href="#coldataplot" class="anchor"></a>Column data plot</h3>
<p>The column data plot can display one or two of the provided column annotations (from the <code>colData</code> slot). Depending on the class of the selected annotations, the panel shows either a Hinton diagram <span class="citation">(Hinton and Shallice 1991; Bremner, Gotts, and Denham 1994)</span>, a violin plot, or a scatter plot.</p>
<p><img src="img/colDataPlot.png"></p>
</div>
<div id="rowdataplot" class="section level3">
<h3 class="hasAnchor">
<a href="#rowdataplot" class="anchor"></a>Row data plot</h3>
<p>Analogous to the <a href="#coldataplot">column data plot</a> above, the row data plot displays one or two of the provided row annotations (from the <code>rowData</code> slot). Depending on the class of the selected annotations, the panel displays either a Hinton plot, a violin plot, or a scatter plot.</p>
<p><img src="img/rowDataPlot.png"></p>
</div>
<div id="heatmap" class="section level3">
<h3 class="hasAnchor">
<a href="#heatmap" class="anchor"></a>Heat map</h3>
<p>The heat map panel displays, for any assay, the observed values for a subset of the features across the samples.</p>
<p><img src="img/heatMap.png"></p>
</div>
<div id="featassayplot" class="section level3">
<h3 class="hasAnchor">
<a href="#featassayplot" class="anchor"></a>Feature assay plot</h3>
<p>The feature assay plot displays the observed values for one feature across the samples. It is also possible to plot the observed values for two features, in a scatter plot.</p>
<p><img src="img/featAssayPlot.png"></p>
</div>
<div id="sampassayplot" class="section level3">
<h3 class="hasAnchor">
<a href="#sampassayplot" class="anchor"></a>Sample assay plot</h3>
<p>Analogous to the <a href="#featassayplot">feature assay plot</a> above, the sample assay plot shows the observed values for all features, for one of the samples. It is also possible to plot the observed values for two samples, in a scatter plot.</p>
<p><img src="img/sampAssayPlot.png"></p>
</div>
<div id="rowstattable" class="section level3">
<h3 class="hasAnchor">
<a href="#rowstattable" class="anchor"></a>Row statistics table</h3>
<p>The row statistics table displays all information provided in the <code>rowData</code> slot of the <code>SummarizedExperiment</code> object, leveraging the interactivity provided by the <em><a href="https://CRAN.R-project.org/package=DT">DT</a></em> package.</p>
<p><img src="img/rowStatTable.png"></p>
</div>
<div id="colstattable" class="section level3">
<h3 class="hasAnchor">
<a href="#colstattable" class="anchor"></a>Column statistics table</h3>
<p>Analogous to the <a href="#rowstattable">row statistics table</a> above, the column statistics table displays all information provided in the <code>colData</code> slot of the <code>SummarizedExperiment</code> object.</p>
<p><img src="img/colStatTable.png"></p>
</div>
<div id="data-parameters-box" class="section level3">
<h3 class="hasAnchor">
<a href="#data-parameters-box" class="anchor"></a>‘Data parameters’ collapsible box</h3>
<p>Each plot panel type has a <code>Data parameters</code> collapsible box. This box has different content for the each panel types, but in all cases it lets the user control the data that is displayed in the plot.</p>
<p><img src="img/colDataPlotDataParams.png"></p>
</div>
<div id="visual-parameters-box" class="section level3">
<h3 class="hasAnchor">
<a href="#visual-parameters-box" class="anchor"></a>‘Visual parameters’ collapsible box</h3>
<p>In contrast to the <code>Data parameters</code> collapsible box that lets users control <em>what</em> is displayed in the plot, the <code>Visual parameters</code> box lets users control <em>how</em> the information is displayed.</p>
<p>This collapsible box contains the controls to change the size, shape, opacity, and color of the points, to facet the plot by any available categorical annotation, to subsample points for increased speed of plot rendering, and to control how legends are displayed.</p>
<p><img src="img/colDataPlotVisualParams.png"></p>
</div>
<div id="selection-parameter-box" class="section level3">
<h3 class="hasAnchor">
<a href="#selection-parameter-box" class="anchor"></a>‘Selection parameters’ collapsible box</h3>
<p>The <code>Selection parameters</code> collapsible box provides controls to transfer selections of points (features or samples) between panels.</p>
<p>We discuss point transmission in more detail <a href="#transmit-point-selections">later in this workshop</a>.</p>
<!-- TODO: Link "later" to the appropriate section when it exists -->
<p><img src="img/redDimPlotSelectionParams.png"></p>
</div>
<div id="additional-controls" class="section level3">
<h3 class="hasAnchor">
<a href="#additional-controls" class="anchor"></a>Additional controls</h3>
<p>At the top-right corner of the iSEE application, users can find additional controls for reproducibility, configuration, and help. Each of these is discussed in more detail later in this workshop.</p>
<table class="table"><tbody>
<tr class="odd">
<td>Panel organization</td>
<td align="center"><img width="275" src="img/PanelOrganization.png"></td>
<td><a href="#configuration-ui">Organize panels</a></td>
</tr>
<tr class="even">
<td>Diagnostics</td>
<td align="center"><img width="275" src="img/Diagnostics.png"></td>
<td>
<a href="#transmit-point-selections">Examine panel chart</a>, <a href="#reproducibility">Extract R code</a>, <a href="#configuration-ui">Display panel settings</a>
</td>
</tr>
<tr class="odd">
<td>Documentation</td>
<td align="center"><img width="275" src="img/Documentation.png"></td>
<td>
<a href="#sharing-tours">Quick tour</a>, <a href="#more-resources">Open the vignette</a>
</td>
</tr>
<tr class="even">
<td>Additional info</td>
<td align="center"><img width="275" src="img/additionalInfo.png"></td>
<td>
<a href="#reproducibility">About this session</a>, <a href="#more-resources">About iSEE</a>
</td>
</tr>
</tbody></table>
<!-- Show how to change what is plotted, faceting, changing appearance/color --><!-- Transmit selections --><!-- Subsampling -->
</div>
</div>
<div id="configuration-ui" class="section level2">
<h2 class="hasAnchor">
<a href="#configuration-ui" class="anchor"></a>Configuration of the app interface</h2>
<p>As mentioned above, the default behaviour of the <code><a href="https://www.rdocumentation.org/packages/iSEE/topics/iSEE">iSEE()</a></code> function is to launch an instance of the user interface that displays one panel of each of the standard types (provided the underlying data is available, e.g. for reduced dimension plots the <code>reducedDim</code> slot is required). However, in some cases it is desirable to have multiple panels of the same type, and/or exclude some panel types.</p>
<p>In order to accommodate such situations, users can add, remove, change the order of and resize all panels via the <code>Panel organization</code> menu in the top-right corner.</p>
<blockquote>
<p><i class="fa fa-wrench fa-2x"></i> <strong>Action:</strong> Using your open <em><a href="https://bioconductor.org/packages/3.10/iSEE">iSEE</a></em> instance, open the <code>Organization</code> menu and click on the <code>Organize panels</code> button. Try to add and remove panels, and resize the existing ones. Remember to click on <code>Apply settings</code> to apply the changes you have made to the interface.</p>
</blockquote>
<p>Clicking in the selectize box listing all current panels, will present you with a drop-down menu from which you can choose additional panels to add. Similarly, panels can be removed by clicking on the <i class="fa fa-times"></i> icon associated with the panel name.</p>
<p>Each panel can be individually resized by changing the width and height. Note that the total width of a row in the interface is 12 units. When the width of a panel is greater than the space available, the panel is moved to a new row.</p>
<p>It can take a great amount of time to achieve a satisfactory panel configuration. To avoid the need to manually organize the panels each time the app is opened, <em><a href="https://bioconductor.org/packages/3.10/iSEE">iSEE</a></em> offers the possibility to export the current panel settings.</p>
<blockquote>
<p><i class="fa fa-wrench fa-2x"></i> <strong>Action:</strong> click on the wrench icon (<i class="fa fa-wrench"></i>) in the top-right corner, and select ‘Display panel settings’. Scroll to the end of the code in the popup window and copy the definition of the <code>initialPanels</code> variable.</p>
</blockquote>
<p>For example:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1">initialPanels &lt;-<span class="st"> </span><span class="kw">DataFrame</span>(</a>
<a class="sourceLine" id="cb13-2" data-line-number="2">    <span class="dt">Name=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"Reduced dimension plot 1"</span>, <span class="st">"Column data plot 1"</span>, </a>
<a class="sourceLine" id="cb13-3" data-line-number="3">           <span class="st">"Row data plot 1"</span>, <span class="st">"Heat map 1"</span>, </a>
<a class="sourceLine" id="cb13-4" data-line-number="4">           <span class="st">"Feature assay plot 1"</span>, <span class="st">"Sample assay plot 1"</span>, </a>
<a class="sourceLine" id="cb13-5" data-line-number="5">           <span class="st">"Row statistics table 1"</span>, <span class="st">"Column statistics table 1"</span></a>
<a class="sourceLine" id="cb13-6" data-line-number="6">        ),</a>
<a class="sourceLine" id="cb13-7" data-line-number="7">    <span class="dt">Width=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(3L, 3L, 3L, 3L, 3L, 3L, 3L, 3L),</a>
<a class="sourceLine" id="cb13-8" data-line-number="8">    <span class="dt">Height=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(460L, 460L, 460L, 460L, 460L, 460L, 460L, 460L)</a>
<a class="sourceLine" id="cb13-9" data-line-number="9">)</a></code></pre></div>
<blockquote>
<p><i class="fa fa-wrench fa-2x"></i> <strong>Action:</strong> Close the application, run the code that you just copied in your <em>R</em> session, and relaunch the application, now with the additional argument <code>initialPanels</code> set:</p>
</blockquote>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="cf">if</span> (<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/interactive">interactive</a></span>()) {</a>
<a class="sourceLine" id="cb14-2" data-line-number="2">    <span class="kw"><a href="https://www.rdocumentation.org/packages/iSEE/topics/iSEE">iSEE</a></span>(sce, <span class="dt">initialPanels =</span> initialPanels)</a>
<a class="sourceLine" id="cb14-3" data-line-number="3">}</a></code></pre></div>
<p>Notice how the panel configuration at startup is now identical to the one you saw just before displaying the panel settings.</p>
<p>Now that we have an application consisting of a suitable collection of panels for our data set, we’ll explore the collapsible boxes below each plot, which lets us control their appearance.</p>
<div id="choose-reduced-dimension" class="section level3">
<h3 class="hasAnchor">
<a href="#choose-reduced-dimension" class="anchor"></a>Displaying a different reduced dimension representation</h3>
<p>By default, each <em>Reduced dimension plot</em> displays the first reduced dimension representation provided (contained in the <code>reducedDim</code> slot of the <code>SingleCellExperiment</code> object). If additional reduced dimension representations are included, the one chosen for the display can be changed in the <code>Data parameters</code> collapsible box in the <em>Reduced dimension plot</em> panel.</p>
<blockquote>
<p><i class="fa fa-wrench fa-2x"></i> <strong>Action:</strong> Use this control to show the t-SNE representation instead of the PCA.</p>
</blockquote>
<p>Notably, the instance can be preconfigured to immediately display the t-SNE representation when the application is launched.</p>
<blockquote>
<p><i class="fa fa-wrench fa-2x"></i> <strong>Action:</strong> Close the application and run the following block of code to show two <em>Reduced dimension plot</em> panels; one showing the PCA representation, one showing the t-SNE representation.</p>
</blockquote>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1">redDimArgs &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/iSEE/topics/redDimPlotDefaults">redDimPlotDefaults</a></span>(sce, <span class="dv">2</span>)</a>
<a class="sourceLine" id="cb15-2" data-line-number="2">redDimArgs[<span class="dv">1</span>, <span class="st">"Type"</span>] &lt;-<span class="st"> "PCA"</span></a>
<a class="sourceLine" id="cb15-3" data-line-number="3">redDimArgs[<span class="dv">2</span>, <span class="st">"Type"</span>] &lt;-<span class="st"> "TSNE"</span></a>
<a class="sourceLine" id="cb15-4" data-line-number="4">initialPanels &lt;-<span class="st"> </span><span class="kw">DataFrame</span>(</a>
<a class="sourceLine" id="cb15-5" data-line-number="5">    <span class="dt">Name=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"Reduced dimension plot 1"</span>, <span class="st">"Reduced dimension plot 2"</span>),</a>
<a class="sourceLine" id="cb15-6" data-line-number="6">    <span class="dt">Width=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(4L, 4L),</a>
<a class="sourceLine" id="cb15-7" data-line-number="7">    <span class="dt">Height=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">400</span>, <span class="dv">400</span>)</a>
<a class="sourceLine" id="cb15-8" data-line-number="8">)</a>
<a class="sourceLine" id="cb15-9" data-line-number="9"><span class="cf">if</span> (<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/interactive">interactive</a></span>()) {</a>
<a class="sourceLine" id="cb15-10" data-line-number="10">    <span class="kw"><a href="https://www.rdocumentation.org/packages/iSEE/topics/iSEE">iSEE</a></span>(sce, <span class="dt">redDimArgs =</span> redDimArgs, <span class="dt">initialPanels =</span> initialPanels)</a>
<a class="sourceLine" id="cb15-11" data-line-number="11">}</a></code></pre></div>
<p><img src="img/ReducedDimensionType.png"></p>
</div>
<div id="color-by-annotation" class="section level3">
<h3 class="hasAnchor">
<a href="#color-by-annotation" class="anchor"></a>Coloring points by row/column annotation</h3>
<p>With default settings, all points are colored black. However, any provided row (for features) or column (for samples) annotation can be used to color the points.</p>
<blockquote>
<p><i class="fa fa-wrench fa-2x"></i> <strong>Action:</strong> To demonstrate this, open the <code>Visual parameters</code> collapsible box in one of the <em>Reduced dimension plot</em> panels, and set <code>Color by:</code> to <code>Column data</code>. Scroll through the dropdown menu that appears to see all the sample annotations that can be used to color the points. All these values are taken from the <code>colData</code> slot of the provided object.</p>
</blockquote>
<blockquote>
<p><i class="fa fa-wrench fa-2x"></i> <strong>Action:</strong> First, color the cells by <code>Cluster</code>, which contains the cluster labels that were assigned to the cells in the preprocessing step <a href="#prepare-scrnaseq-data">in a previous section</a>. Note how a discrete color scheme is selected, since the variable is categorical.</p>
</blockquote>
<blockquote>
<p><i class="fa fa-wrench fa-2x"></i> <strong>Action:</strong> Next, color the cells by the log10-transformed number of detected genes (<code>log10_total_features_by_counts</code>). Notice how the color scale is now continuous.</p>
</blockquote>
<p>The instance can be preconfigured to control the coloring of data points, to display the same result on startup.</p>
<blockquote>
<p><i class="fa fa-wrench fa-2x"></i> <strong>Action:</strong> Close the application and run the following block of code to continue editing the configuration of the two <em>Reduced dimension plot</em> panels prepared in the <a href="#choose-reduced-dimension">previous section</a>.</p>
</blockquote>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1">redDimArgs[[<span class="st">"ColorBy"</span>]] &lt;-<span class="st"> "Column data"</span></a>
<a class="sourceLine" id="cb16-2" data-line-number="2">redDimArgs[<span class="dv">1</span>, <span class="st">"ColorByColData"</span>] &lt;-<span class="st"> "log10_total_features_by_counts"</span></a>
<a class="sourceLine" id="cb16-3" data-line-number="3">redDimArgs[<span class="dv">2</span>, <span class="st">"ColorByColData"</span>] &lt;-<span class="st"> "Cluster"</span></a>
<a class="sourceLine" id="cb16-4" data-line-number="4">initialPanels &lt;-<span class="st"> </span><span class="kw">DataFrame</span>(</a>
<a class="sourceLine" id="cb16-5" data-line-number="5">    <span class="dt">Name=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"Reduced dimension plot 1"</span>, <span class="st">"Reduced dimension plot 2"</span>),</a>
<a class="sourceLine" id="cb16-6" data-line-number="6">    <span class="dt">Width=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(4L, 4L),</a>
<a class="sourceLine" id="cb16-7" data-line-number="7">    <span class="dt">Height=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">400</span>, <span class="dv">400</span>)</a>
<a class="sourceLine" id="cb16-8" data-line-number="8">)</a>
<a class="sourceLine" id="cb16-9" data-line-number="9"><span class="cf">if</span> (<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/interactive">interactive</a></span>()) {</a>
<a class="sourceLine" id="cb16-10" data-line-number="10">    <span class="kw"><a href="https://www.rdocumentation.org/packages/iSEE/topics/iSEE">iSEE</a></span>(sce, <span class="dt">redDimArgs =</span> redDimArgs, <span class="dt">initialPanels =</span> initialPanels)</a>
<a class="sourceLine" id="cb16-11" data-line-number="11">}</a></code></pre></div>
<p><img src="img/ColorByColData.png"></p>
<p><em><a href="https://bioconductor.org/packages/3.10/iSEE">iSEE</a></em> allows precise control over the color schemes used. More information about this can be found in the <a href="https://bioconductor.org/packages/3.10/iSEE/vignettes/ecm.html">ExperimentColorMap</a> vignette.</p>
</div>
<div id="color-by-feature" class="section level3">
<h3 class="hasAnchor">
<a href="#color-by-feature" class="anchor"></a>Coloring points by feature value</h3>
<p>In addition to coloring cells by the provided annotations as above, we can also color them by the observed values for a given feature (as provided in one of the <code>assays</code>).</p>
<blockquote>
<p><i class="fa fa-wrench fa-2x"></i> <strong>Action:</strong> Set <code>Color by:</code> to <code>Feature name</code> in one of the <em>Reduced dimension plot</em> panels, and select one of the genes from the dropdown menu. You can search for a gene of interest by typing in the dropdown box. You can also choose which assay should be used to extract the values to color according to.</p>
</blockquote>
<p>Again, the same result can be achieved by preconfiguring the application as follows.</p>
<blockquote>
<p><i class="fa fa-wrench fa-2x"></i> <strong>Action:</strong> Close the application and run the following block of code to set up 3 panels, all displaying the t-SNE representation. In the first panel, we color cells by their <code>Cluster</code> label. In the second and third panel, we display <em>CD79A</em> and <em>CD74</em>, a combination of gene markers typically observed in B cells.</p>
</blockquote>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1">## The displayed features IDs must be row names of the object</a>
<a class="sourceLine" id="cb17-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"CD79A"</span>, <span class="st">"CD74"</span>) <span class="op">%in%</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(sce)</a>
<a class="sourceLine" id="cb17-3" data-line-number="3"><span class="co">#&gt; [1] TRUE TRUE</span></a>
<a class="sourceLine" id="cb17-4" data-line-number="4"></a>
<a class="sourceLine" id="cb17-5" data-line-number="5">redDimArgs &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/iSEE/topics/redDimPlotDefaults">redDimPlotDefaults</a></span>(sce, <span class="dv">3</span>)</a>
<a class="sourceLine" id="cb17-6" data-line-number="6">redDimArgs[[<span class="st">"Type"</span>]] &lt;-<span class="st"> "TSNE"</span></a>
<a class="sourceLine" id="cb17-7" data-line-number="7">redDimArgs[<span class="dv">1</span>, <span class="st">"ColorBy"</span>] &lt;-<span class="st"> "Column data"</span></a>
<a class="sourceLine" id="cb17-8" data-line-number="8">redDimArgs[<span class="dv">1</span>, <span class="st">"ColorByColData"</span>] &lt;-<span class="st"> "Cluster"</span></a>
<a class="sourceLine" id="cb17-9" data-line-number="9">redDimArgs[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>, <span class="st">"ColorBy"</span>] &lt;-<span class="st"> "Feature name"</span></a>
<a class="sourceLine" id="cb17-10" data-line-number="10">redDimArgs[<span class="dv">2</span>, <span class="st">"ColorByFeatName"</span>] &lt;-<span class="st"> "CD79A"</span></a>
<a class="sourceLine" id="cb17-11" data-line-number="11">redDimArgs[<span class="dv">3</span>, <span class="st">"ColorByFeatName"</span>] &lt;-<span class="st"> "CD74"</span></a>
<a class="sourceLine" id="cb17-12" data-line-number="12">initialPanels &lt;-<span class="st"> </span><span class="kw">DataFrame</span>(</a>
<a class="sourceLine" id="cb17-13" data-line-number="13">    <span class="dt">Name=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste</a></span>(<span class="st">"Reduced dimension plot"</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>),</a>
<a class="sourceLine" id="cb17-14" data-line-number="14">    <span class="dt">Width=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rep">rep</a></span>(4L, 3L),</a>
<a class="sourceLine" id="cb17-15" data-line-number="15">    <span class="dt">Height=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rep">rep</a></span>(<span class="dv">400</span>, 3L)</a>
<a class="sourceLine" id="cb17-16" data-line-number="16">)</a>
<a class="sourceLine" id="cb17-17" data-line-number="17"><span class="cf">if</span> (<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/interactive">interactive</a></span>()) {</a>
<a class="sourceLine" id="cb17-18" data-line-number="18">    <span class="kw"><a href="https://www.rdocumentation.org/packages/iSEE/topics/iSEE">iSEE</a></span>(sce, <span class="dt">redDimArgs =</span> redDimArgs, <span class="dt">initialPanels =</span> initialPanels)</a>
<a class="sourceLine" id="cb17-19" data-line-number="19">}</a></code></pre></div>
<p><img src="img/ColorByFeature.png"></p>
</div>
<div id="highlight" class="section level3">
<h3 class="hasAnchor">
<a href="#highlight" class="anchor"></a>Highlighting a single point</h3>
<p>The <code>Color by: Sample name</code> option in the <em>Reduced dimension plot</em> can also be used to highlight a single sample.</p>
<blockquote>
<p><i class="fa fa-wrench fa-2x"></i> <strong>Action:</strong> Set <code>Color by:</code> to <code>Sample name</code> in the <em>Reduced dimension plot</em>, and use the dropdown menu underneath to change the highlighted cell.</p>
</blockquote>
<p>Again, the same result can be achieved by preconfiguring the application as follows. Note that it is more robust to provide the numerical index of the sample in the <code>sce</code> object (see the <code>match</code> statements in the code chunk below).</p>
<blockquote>
<p><i class="fa fa-wrench fa-2x"></i> <strong>Action:</strong> Close the app and run the following block of code to continue editing the configuration of the two <em>Reduced dimension plot</em> panels prepared in the <a href="#color-by-feature">previous section</a>. We also open the <em>Visual parameters</em> collapsible box to highlight the preconfigured options.</p>
</blockquote>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1">redDimArgs[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>, <span class="st">"ColorBy"</span>] &lt;-<span class="st"> "Sample name"</span></a>
<a class="sourceLine" id="cb18-2" data-line-number="2">redDimArgs[<span class="dv">2</span>, <span class="st">"ColorBySampName"</span>] &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/match">match</a></span>(<span class="st">"Cell2439"</span>, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(sce))</a>
<a class="sourceLine" id="cb18-3" data-line-number="3">redDimArgs[<span class="dv">3</span>, <span class="st">"ColorBySampName"</span>] &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/match">match</a></span>(<span class="st">"Cell673"</span>, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(sce))</a>
<a class="sourceLine" id="cb18-4" data-line-number="4">redDimArgs[[<span class="st">"VisualBoxOpen"</span>]] &lt;-<span class="st"> </span><span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb18-5" data-line-number="5">initialPanels &lt;-<span class="st"> </span><span class="kw">DataFrame</span>(</a>
<a class="sourceLine" id="cb18-6" data-line-number="6">    <span class="dt">Name=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste</a></span>(<span class="st">"Reduced dimension plot"</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>),</a>
<a class="sourceLine" id="cb18-7" data-line-number="7">    <span class="dt">Width=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rep">rep</a></span>(4L, 3L),</a>
<a class="sourceLine" id="cb18-8" data-line-number="8">    <span class="dt">Height=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rep">rep</a></span>(<span class="dv">400</span>, 3L)</a>
<a class="sourceLine" id="cb18-9" data-line-number="9">)</a>
<a class="sourceLine" id="cb18-10" data-line-number="10"><span class="cf">if</span> (<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/interactive">interactive</a></span>()) {</a>
<a class="sourceLine" id="cb18-11" data-line-number="11">    <span class="kw"><a href="https://www.rdocumentation.org/packages/iSEE/topics/iSEE">iSEE</a></span>(sce, <span class="dt">redDimArgs =</span> redDimArgs, <span class="dt">initialPanels =</span> initialPanels)</a>
<a class="sourceLine" id="cb18-12" data-line-number="12">}</a></code></pre></div>
<p><img src="img/ColorBySample.png"></p>
</div>
<div id="change-shape-opacity" class="section level3">
<h3 class="hasAnchor">
<a href="#change-shape-opacity" class="anchor"></a>Changing the size, shape and opacity of points</h3>
<p>By default, only the <code>Color</code> options are displayed in the <code>Visual parameters</code> box. Additional controls can be shown by clicking the check boxes for <code>Shape</code>, <code>Points</code>, <code>Facets</code> and <code>Other</code>. The <code>Shape</code> and <code>Points</code> panels let you set the shape and size of points according to provided annotations (or to resize all points), and to set the point opacity. Here, you can also downsample points to increase the speed of rendering, which can be useful for data sets with large numbers of points.</p>
<blockquote>
<p><i class="fa fa-wrench fa-2x"></i> <strong>Action:</strong> Close the app and run the following block of code to continue editing the configuration of the two <em>Reduced dimension plot</em> panels prepared in the <a href="#color-by-feature">previous section</a>. Here, we reduce the point size and increase the transparency (i.e. reduce the value for the alpha attribute). We also apply a downsampling grid of 100 horizontal and vertical bins to the plot, where only a single data point is plotted (if any) for each bin. Finally, we display the <code>Shape</code> and <code>Points</code> options.</p>
</blockquote>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1">redDimArgs[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>, <span class="st">"PointSize"</span>] &lt;-<span class="st"> </span><span class="fl">0.4</span></a>
<a class="sourceLine" id="cb19-2" data-line-number="2">redDimArgs[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>, <span class="st">"PointAlpha"</span>] &lt;-<span class="st"> </span><span class="fl">0.4</span></a>
<a class="sourceLine" id="cb19-3" data-line-number="3">redDimArgs[[<span class="st">"VisualBoxOpen"</span>]] &lt;-<span class="st"> </span><span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb19-4" data-line-number="4">redDimArgs[[<span class="st">"Downsample"</span>]] &lt;-<span class="st"> </span><span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb19-5" data-line-number="5">redDimArgs[[<span class="st">"SampleRes"</span>]] &lt;-<span class="st"> </span><span class="dv">200</span></a>
<a class="sourceLine" id="cb19-6" data-line-number="6">redDimArgs[[<span class="st">"VisualChoices"</span>]][[<span class="dv">2</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(<span class="st">"Shape"</span>, <span class="st">"Points"</span>)</a>
<a class="sourceLine" id="cb19-7" data-line-number="7">redDimArgs[[<span class="st">"VisualChoices"</span>]][[<span class="dv">3</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(<span class="st">"Shape"</span>, <span class="st">"Points"</span>)</a>
<a class="sourceLine" id="cb19-8" data-line-number="8">initialPanels &lt;-<span class="st"> </span><span class="kw">DataFrame</span>(</a>
<a class="sourceLine" id="cb19-9" data-line-number="9">    <span class="dt">Name=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste</a></span>(<span class="st">"Reduced dimension plot"</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>),</a>
<a class="sourceLine" id="cb19-10" data-line-number="10">    <span class="dt">Width=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rep">rep</a></span>(4L, 3L),</a>
<a class="sourceLine" id="cb19-11" data-line-number="11">    <span class="dt">Height=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rep">rep</a></span>(<span class="dv">400</span>, 3L)</a>
<a class="sourceLine" id="cb19-12" data-line-number="12">)</a>
<a class="sourceLine" id="cb19-13" data-line-number="13"><span class="cf">if</span> (<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/interactive">interactive</a></span>()) {</a>
<a class="sourceLine" id="cb19-14" data-line-number="14">    <span class="kw"><a href="https://www.rdocumentation.org/packages/iSEE/topics/iSEE">iSEE</a></span>(sce, <span class="dt">redDimArgs =</span> redDimArgs, <span class="dt">initialPanels =</span> initialPanels)</a>
<a class="sourceLine" id="cb19-15" data-line-number="15">}</a></code></pre></div>
<p><img src="img/ConfigureVisualParameters.png"></p>
</div>
<div id="change-metadata-displayed" class="section level3">
<h3 class="hasAnchor">
<a href="#change-metadata-displayed" class="anchor"></a>Changing the displayed metadata</h3>
<p>The <em>Column data plot</em> and <em>Row data plot</em> panels are used to display the information in the <code>colData</code> and <code>rowData</code> slots of the <code>SummarizedExperiment</code> object. These panels can display either one or two annotations. By default, the first annotation in the respective metadata table is selected.</p>
<p>The plot layout adapts automatically to the type of annotation(s) displayed.</p>
<ul>
<li>When all chosen annotations are categorical, <em><a href="https://bioconductor.org/packages/3.10/iSEE">iSEE</a></em> displays a Hinton plot, consisting of rectangles with an area proportional to the number of observations they represent.</li>
<li>For a single continuous variable, or one continuous and one categorical variable, <em><a href="https://bioconductor.org/packages/3.10/iSEE">iSEE</a></em> uses violin plot(s).</li>
<li>If two continuous variables are chosen, <em><a href="https://bioconductor.org/packages/3.10/iSEE">iSEE</a></em> displays a scatter plot.</li>
</ul>
<p>As a side note, categorical variables with “too many” unique values (which is typically the case for sample IDs, for instance) are represented as numeric variables, by replacing each value by an index. The limit for what is considered “too many” unique values can be set using <code><a href="https://www.rdocumentation.org/packages/base/topics/options">options(iSEE.maxlevels = 30)</a></code>.</p>
<blockquote>
<p><i class="fa fa-wrench fa-2x"></i> <strong>Action:</strong> Explore the different types of plots provided in the Column data or Row data plot panels by selecting different annotations as the <code>Column of interest (Y-axis)</code> and/or setting <code>X-axis:</code> to <code>Column data</code> and selecting an additional annotation.</p>
</blockquote>
<p>In the following code chunk we demonstrate the preconfiguration of an app instance that immediately displays two panels.</p>
<blockquote>
<p><i class="fa fa-wrench fa-2x"></i> <strong>Action:</strong> Close the app and run the following block of code to set up one <em>Column data plot</em> and one <em>Row data plot</em> panel. We set up the <em>Column data plot</em> panel to immediately examine the proportion of counts assigned to mitochondrial features in each cluster. We set up the <em>Row data plot</em> panel to examine the proportion of cells with detectable expression against the log-transformed total counts for each gene.</p>
</blockquote>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1">colDataArgs &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/iSEE/topics/colDataPlotDefaults">colDataPlotDefaults</a></span>(sce, <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb20-2" data-line-number="2">colDataArgs[<span class="dv">1</span>, <span class="st">"YAxis"</span>] &lt;-<span class="st"> "pct_counts_MT"</span></a>
<a class="sourceLine" id="cb20-3" data-line-number="3">colDataArgs<span class="op">$</span>XAxis &lt;-<span class="st"> "Column data"</span></a>
<a class="sourceLine" id="cb20-4" data-line-number="4">colDataArgs[<span class="dv">1</span>, <span class="st">"XAxisColData"</span>] &lt;-<span class="st"> "Cluster"</span></a>
<a class="sourceLine" id="cb20-5" data-line-number="5">rowDataArgs &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/iSEE/topics/rowDataPlotDefaults">rowDataPlotDefaults</a></span>(sce, <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb20-6" data-line-number="6">rowDataArgs[<span class="dv">1</span>, <span class="st">"YAxis"</span>] &lt;-<span class="st"> "n_cells_by_counts"</span></a>
<a class="sourceLine" id="cb20-7" data-line-number="7">rowDataArgs<span class="op">$</span>XAxis &lt;-<span class="st"> "Row data"</span></a>
<a class="sourceLine" id="cb20-8" data-line-number="8">rowDataArgs[<span class="dv">1</span>, <span class="st">"XAxisRowData"</span>] &lt;-<span class="st"> "log10_total_counts"</span></a>
<a class="sourceLine" id="cb20-9" data-line-number="9">initialPanels &lt;-<span class="st"> </span><span class="kw">DataFrame</span>(</a>
<a class="sourceLine" id="cb20-10" data-line-number="10">    <span class="dt">Name=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"Column data plot 1"</span>, <span class="st">"Row data plot 1"</span>),</a>
<a class="sourceLine" id="cb20-11" data-line-number="11">    <span class="dt">Width=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rep">rep</a></span>(4L, 2L),</a>
<a class="sourceLine" id="cb20-12" data-line-number="12">    <span class="dt">Height=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rep">rep</a></span>(<span class="dv">400</span>, 2L)</a>
<a class="sourceLine" id="cb20-13" data-line-number="13">)</a>
<a class="sourceLine" id="cb20-14" data-line-number="14"><span class="cf">if</span> (<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/interactive">interactive</a></span>()) {</a>
<a class="sourceLine" id="cb20-15" data-line-number="15">    <span class="kw"><a href="https://www.rdocumentation.org/packages/iSEE/topics/iSEE">iSEE</a></span>(sce, <span class="dt">colDataArgs =</span> colDataArgs, <span class="dt">rowDataArgs =</span> rowDataArgs, <span class="dt">initialPanels =</span> initialPanels)</a>
<a class="sourceLine" id="cb20-16" data-line-number="16">}</a></code></pre></div>
<p><img src="img/ConfigureDataPlotPanels.png"></p>
</div>
<div id="display-gene-expression" class="section level3">
<h3 class="hasAnchor">
<a href="#display-gene-expression" class="anchor"></a>Displaying gene expression</h3>
<p>The <em>Feature assay plot</em> is used to display the observed values for one feature across the samples on the Y-axis. By default, the first feature in the <code>rownames</code> of the dataset is selected, displaying its distribution across the entire dataset.</p>
<p>While continuous measurements are more common (e.g., gene expression), discrete assays are also supported (e.g., single nucleotide polymorphisms), and displayed as a Hinton plot, consisting of rectangles with an area proportional to the number of samples they represent.</p>
<p>It is also possible to plot the observed values a second feature on the X-axis. This will result in a scatter plot or Hinton plot, depending on the continuous or discrete nature of the data, respectively.</p>
<p>Alternatively, the X-axis can also be used to stratify gene expression according to a discrete sample metadata in the form of a violin plot, while a continuous metadata will generate a scatter plot.</p>
<blockquote>
<p><i class="fa fa-wrench fa-2x"></i> <strong>Action:</strong> Explore the different types of plots generated by the Feature assay plot panel using the choices available for the two axes.</p>
</blockquote>
<p>Similarly to all other panels, <em>Feature assay plot</em> can be preconfigured to immediately display specific information when the app is launched.</p>
<blockquote>
<p><i class="fa fa-wrench fa-2x"></i> <strong>Action:</strong> Close the app and run the following block of code to set up three <em>Feature assay plot</em> panels demonstrating different configurations. All three panels display the same gene on the Y-axis and color data points according to their cluster label. On the X-axis, the first panel displays a discrete sample metadata variable, the cluster label; the second panel display a continuous sample metadata variable, the log-transformed library size; the third panel displays the expression of a second gene.</p>
</blockquote>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1">featAssayArgs &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/iSEE/topics/featAssayPlotDefaults">featAssayPlotDefaults</a></span>(sce, <span class="dv">3</span>)</a>
<a class="sourceLine" id="cb21-2" data-line-number="2">featAssayArgs[[<span class="st">"ColorBy"</span>]] &lt;-<span class="st"> "Column data"</span></a>
<a class="sourceLine" id="cb21-3" data-line-number="3">featAssayArgs[[<span class="st">"ColorByColData"</span>]] &lt;-<span class="st"> "Cluster"</span></a>
<a class="sourceLine" id="cb21-4" data-line-number="4">featAssayArgs[[<span class="st">"YAxisFeatName"</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/match">match</a></span>(<span class="st">"CD3D"</span>, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(sce))</a>
<a class="sourceLine" id="cb21-5" data-line-number="5">featAssayArgs[<span class="dv">1</span>, <span class="st">"XAxis"</span>] &lt;-<span class="st"> "Column data"</span></a>
<a class="sourceLine" id="cb21-6" data-line-number="6">featAssayArgs[<span class="dv">1</span>, <span class="st">"XAxisColData"</span>] &lt;-<span class="st"> "Cluster"</span></a>
<a class="sourceLine" id="cb21-7" data-line-number="7">featAssayArgs[<span class="dv">2</span>, <span class="st">"XAxis"</span>] &lt;-<span class="st"> "Column data"</span></a>
<a class="sourceLine" id="cb21-8" data-line-number="8">featAssayArgs[<span class="dv">2</span>, <span class="st">"XAxisColData"</span>] &lt;-<span class="st"> "log10_total_counts"</span></a>
<a class="sourceLine" id="cb21-9" data-line-number="9">featAssayArgs[<span class="dv">3</span>, <span class="st">"XAxis"</span>] &lt;-<span class="st"> "Feature name"</span></a>
<a class="sourceLine" id="cb21-10" data-line-number="10">featAssayArgs[<span class="dv">3</span>, <span class="st">"XAxisFeatName"</span>] &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/match">match</a></span>(<span class="st">"CD40LG"</span>, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(sce))</a>
<a class="sourceLine" id="cb21-11" data-line-number="11">featAssayArgs[[<span class="st">"DataBoxOpen"</span>]] &lt;-<span class="st"> </span><span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb21-12" data-line-number="12">initialPanels &lt;-<span class="st"> </span><span class="kw">DataFrame</span>(</a>
<a class="sourceLine" id="cb21-13" data-line-number="13">    <span class="dt">Name=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste</a></span>(<span class="st">"Feature assay plot"</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>),</a>
<a class="sourceLine" id="cb21-14" data-line-number="14">    <span class="dt">Width=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rep">rep</a></span>(4L, 3L),</a>
<a class="sourceLine" id="cb21-15" data-line-number="15">    <span class="dt">Height=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rep">rep</a></span>(<span class="dv">400</span>, 3L)</a>
<a class="sourceLine" id="cb21-16" data-line-number="16">)</a>
<a class="sourceLine" id="cb21-17" data-line-number="17"><span class="cf">if</span> (<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/interactive">interactive</a></span>()) {</a>
<a class="sourceLine" id="cb21-18" data-line-number="18">    <span class="kw"><a href="https://www.rdocumentation.org/packages/iSEE/topics/iSEE">iSEE</a></span>(sce, <span class="dt">featAssayArgs =</span> featAssayArgs, <span class="dt">initialPanels =</span> initialPanels)</a>
<a class="sourceLine" id="cb21-19" data-line-number="19">}</a></code></pre></div>
<p><img src="img/featAssayPlotDataParams.png"></p>
</div>
<div id="transmit-point-selections" class="section level3">
<h3 class="hasAnchor">
<a href="#transmit-point-selections" class="anchor"></a>Linking panels and transmitting point selections</h3>
<p>When exploring data, it is often useful to be able to select a subset of points and investigate different aspects of their characteristic features. In <em><a href="https://bioconductor.org/packages/3.10/iSEE">iSEE</a></em>, this can be achieved by <em>selecting</em> points in one panel, and <em>transmitting</em> this selection to one or more other panels.</p>
<blockquote>
<p><i class="fa fa-wrench fa-2x"></i> <strong>Action:</strong> Select a set of points in one of the <em>Feature assay plot</em> panels, by drawing a rectangle around them. Then open the <em>Selection parameters</em> collapsible box of one of the other <em>Feature assay plot</em> panels, and select the <em>Feature assay plot</em> panel where you made the point selection from the dropdown menu under <strong>Receive column selection from</strong>. Note how the points corresponding to the cells that you selected in the first panel are highlighted in the receiving panel. You can highlight the points in different ways by changing the <strong>Selection effect</strong>.</p>
</blockquote>
<p>Also the brushing and point selection can be preconfigured. In the example below, we configure three types of panels: a <em>Feature assay plot</em>, a <em>Reduced dimension plot</em>, and a <em>Column data plot</em>. For visual convenience, we color data points in all three panels by the cluster label.</p>
<p>To demonstrate the feature, we define a Shiny brush in the <em>Feature assay plot</em> panel to select cells with detectable levels of the <em>CD3E</em> gene. For the other two panels, pay particular attention to the <code>"SelectByPlot"</code> instruction which declares that they should both receive the data points selected in the <em>Feature assay plot</em> panel. In addition, we specify that the <em>Reduced dimension plot</em> should highlight the selection by applying a transparency effect to <em>unselected</em> data points. Alternatively, we use the <em>Column data plot</em> to demonstrate how the selection can also be highlighted by applying a given color (here, red) to the selected data points.</p>
<blockquote>
<p><i class="fa fa-wrench fa-2x"></i> <strong>Action:</strong> Close the app and run the following block of code to launch the example application described above.</p>
</blockquote>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1">featAssayArgs &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/iSEE/topics/featAssayPlotDefaults">featAssayPlotDefaults</a></span>(sce, <span class="dv">2</span>)</a>
<a class="sourceLine" id="cb22-2" data-line-number="2">featAssayArgs[<span class="dv">1</span>, <span class="st">"YAxisFeatName"</span>] &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/match">match</a></span>(<span class="st">"CD3D"</span>, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(sce))</a>
<a class="sourceLine" id="cb22-3" data-line-number="3">featAssayArgs<span class="op">$</span>BrushData[[<span class="dv">1</span>]] &lt;-<span class="st">  </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(</a>
<a class="sourceLine" id="cb22-4" data-line-number="4">    <span class="dt">xmin =</span> <span class="fl">0.5</span>, <span class="dt">xmax =</span> <span class="fl">12.5</span>, <span class="dt">ymin =</span> <span class="fl">0.5</span>, <span class="dt">ymax =</span> <span class="fl">5.7</span>,</a>
<a class="sourceLine" id="cb22-5" data-line-number="5">    <span class="dt">mapping =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(<span class="dt">x =</span> <span class="st">"X"</span>, <span class="dt">y =</span> <span class="st">"Y"</span>, <span class="dt">group =</span> <span class="st">"GroupBy"</span>),</a>
<a class="sourceLine" id="cb22-6" data-line-number="6">    <span class="dt">direction =</span> <span class="st">"xy"</span>, </a>
<a class="sourceLine" id="cb22-7" data-line-number="7">    <span class="dt">brushId =</span> <span class="st">"featAssayPlot1_Brush"</span>, <span class="dt">outputId =</span> <span class="st">"featAssayPlot1"</span>)</a>
<a class="sourceLine" id="cb22-8" data-line-number="8">featAssayArgs[<span class="dv">1</span>, <span class="st">"XAxis"</span>] &lt;-<span class="st"> "Column data"</span></a>
<a class="sourceLine" id="cb22-9" data-line-number="9">featAssayArgs[<span class="dv">1</span>, <span class="st">"XAxisColData"</span>] &lt;-<span class="st"> "Cluster"</span></a>
<a class="sourceLine" id="cb22-10" data-line-number="10">featAssayArgs[<span class="dv">1</span>, <span class="st">"ColorBy"</span>] &lt;-<span class="st"> "Column data"</span></a>
<a class="sourceLine" id="cb22-11" data-line-number="11">featAssayArgs[<span class="dv">1</span>, <span class="st">"ColorByColData"</span>] &lt;-<span class="st"> "Cluster"</span></a>
<a class="sourceLine" id="cb22-12" data-line-number="12">redDimArgs &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/iSEE/topics/redDimPlotDefaults">redDimPlotDefaults</a></span>(sce, <span class="dv">2</span>)</a>
<a class="sourceLine" id="cb22-13" data-line-number="13">redDimArgs[<span class="dv">1</span>, <span class="st">"SelectByPlot"</span>] &lt;-<span class="st"> "Feature assay plot 1"</span></a>
<a class="sourceLine" id="cb22-14" data-line-number="14">redDimArgs[<span class="dv">1</span>, <span class="st">"Type"</span>] &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/match">match</a></span>(<span class="st">"TSNE"</span>, <span class="kw">reducedDimNames</span>(sce))</a>
<a class="sourceLine" id="cb22-15" data-line-number="15">redDimArgs[<span class="dv">1</span>, <span class="st">"ColorBy"</span>] &lt;-<span class="st"> "Column data"</span></a>
<a class="sourceLine" id="cb22-16" data-line-number="16">redDimArgs[<span class="dv">1</span>, <span class="st">"ColorByColData"</span>] &lt;-<span class="st"> "Cluster"</span></a>
<a class="sourceLine" id="cb22-17" data-line-number="17">redDimArgs[<span class="dv">1</span>, <span class="st">"SelectBoxOpen"</span>] &lt;-<span class="st"> </span><span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb22-18" data-line-number="18">colDataArgs &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/iSEE/topics/colDataPlotDefaults">colDataPlotDefaults</a></span>(sce, <span class="dv">2</span>)</a>
<a class="sourceLine" id="cb22-19" data-line-number="19">colDataArgs[<span class="dv">1</span>, <span class="st">"SelectByPlot"</span>] &lt;-<span class="st"> "Feature assay plot 1"</span></a>
<a class="sourceLine" id="cb22-20" data-line-number="20">colDataArgs[<span class="dv">1</span>, <span class="st">"SelectEffect"</span>] &lt;-<span class="st"> "Color"</span></a>
<a class="sourceLine" id="cb22-21" data-line-number="21">colDataArgs[<span class="dv">1</span>, <span class="st">"YAxis"</span>] &lt;-<span class="st"> "log10_total_counts"</span></a>
<a class="sourceLine" id="cb22-22" data-line-number="22">colDataArgs[<span class="dv">1</span>, <span class="st">"XAxis"</span>] &lt;-<span class="st"> "Column data"</span></a>
<a class="sourceLine" id="cb22-23" data-line-number="23">colDataArgs[<span class="dv">1</span>, <span class="st">"XAxisColData"</span>] &lt;-<span class="st"> "pct_counts_MT"</span></a>
<a class="sourceLine" id="cb22-24" data-line-number="24">colDataArgs[<span class="dv">1</span>, <span class="st">"ColorBy"</span>] &lt;-<span class="st"> "Column data"</span></a>
<a class="sourceLine" id="cb22-25" data-line-number="25">colDataArgs[<span class="dv">1</span>, <span class="st">"ColorByColData"</span>] &lt;-<span class="st"> "Cluster"</span></a>
<a class="sourceLine" id="cb22-26" data-line-number="26">colDataArgs[<span class="dv">1</span>, <span class="st">"SelectBoxOpen"</span>] &lt;-<span class="st"> </span><span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb22-27" data-line-number="27">initialPanels &lt;-<span class="st"> </span><span class="kw">DataFrame</span>(</a>
<a class="sourceLine" id="cb22-28" data-line-number="28">    <span class="dt">Name=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(</a>
<a class="sourceLine" id="cb22-29" data-line-number="29">        <span class="st">"Feature assay plot 1"</span>, <span class="st">"Reduced dimension plot 1"</span>, <span class="st">"Column data plot 1"</span>,</a>
<a class="sourceLine" id="cb22-30" data-line-number="30">        <span class="st">"Feature assay plot 2"</span>, <span class="st">"Reduced dimension plot 2"</span>, <span class="st">"Column data plot 2"</span>),</a>
<a class="sourceLine" id="cb22-31" data-line-number="31">    <span class="dt">Width=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rep">rep</a></span>(4L, 3L),</a>
<a class="sourceLine" id="cb22-32" data-line-number="32">    <span class="dt">Height=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rep">rep</a></span>(<span class="dv">400</span>, 3L)</a>
<a class="sourceLine" id="cb22-33" data-line-number="33">)</a>
<a class="sourceLine" id="cb22-34" data-line-number="34"><span class="cf">if</span> (<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/interactive">interactive</a></span>()) {</a>
<a class="sourceLine" id="cb22-35" data-line-number="35">    <span class="kw"><a href="https://www.rdocumentation.org/packages/iSEE/topics/iSEE">iSEE</a></span>(</a>
<a class="sourceLine" id="cb22-36" data-line-number="36">        sce,</a>
<a class="sourceLine" id="cb22-37" data-line-number="37">        <span class="dt">redDimArgs =</span> redDimArgs, <span class="dt">colDataArgs =</span> colDataArgs, <span class="dt">featAssayArgs =</span> featAssayArgs,</a>
<a class="sourceLine" id="cb22-38" data-line-number="38">        <span class="dt">initialPanels =</span> initialPanels</a>
<a class="sourceLine" id="cb22-39" data-line-number="39">    )</a>
<a class="sourceLine" id="cb22-40" data-line-number="40">}</a></code></pre></div>
<p>Notably, iSEE can display active links transmitting point selections between panel using the “<a href="#additional-controls">Examine panel chart</a>” menu. Visible panels that do not send or receive a selection are also shown.</p>
<p><img src="img/GraphSelectionLinks.png"></p>
</div>
<div id="zoom-subregion" class="section level3">
<h3 class="hasAnchor">
<a href="#zoom-subregion" class="anchor"></a>Zooming into a plot subregion</h3>
<p>iSEE allows you to zoom into a subregion of a plot, by simply drawing a rectangle around the area of interest and double-clicking inside the rectangle. To zoom back out to the full plot, double-click anywhere in the plot.</p>
<p>To demonstrate the feature, we set up two <em>Reduced dimension plot</em> panels. In the first panel, we draw a Shiny brush to indicate an area of interest in the reduced dimension plot. We preconfigure the second panel to display only that area. In both panels, we color data points by cluster label using the same color map, to help visualize the zoom relationship between the two panels.</p>
<blockquote>
<p><i class="fa fa-wrench fa-2x"></i> <strong>Action:</strong> Close the app and run the following block of code to launch the example application described above.</p>
</blockquote>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1">redDimArgs &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/iSEE/topics/redDimPlotDefaults">redDimPlotDefaults</a></span>(sce, <span class="dv">2</span>)</a>
<a class="sourceLine" id="cb23-2" data-line-number="2">redDimArgs[[<span class="st">"Type"</span>]] &lt;-<span class="st"> "TSNE"</span></a>
<a class="sourceLine" id="cb23-3" data-line-number="3">redDimArgs[[<span class="st">"ColorBy"</span>]] &lt;-<span class="st"> "Column data"</span></a>
<a class="sourceLine" id="cb23-4" data-line-number="4">redDimArgs[[<span class="st">"ColorByColData"</span>]] &lt;-<span class="st"> "Cluster"</span></a>
<a class="sourceLine" id="cb23-5" data-line-number="5">redDimArgs[<span class="dv">1</span>, <span class="st">"PointSize"</span>] &lt;-<span class="st"> </span><span class="dv">1</span></a>
<a class="sourceLine" id="cb23-6" data-line-number="6">redDimArgs[<span class="dv">2</span>, <span class="st">"PointSize"</span>] &lt;-<span class="st"> </span><span class="dv">2</span></a>
<a class="sourceLine" id="cb23-7" data-line-number="7">redDimArgs[[<span class="st">"BrushData"</span>]][[<span class="dv">1</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(</a>
<a class="sourceLine" id="cb23-8" data-line-number="8">    <span class="dt">xmin =</span> <span class="fl">3.4</span>, <span class="dt">xmax =</span> <span class="fl">27.3</span>, <span class="dt">ymin =</span> <span class="fl">0.8</span>, <span class="dt">ymax =</span> <span class="fl">30.0</span>, </a>
<a class="sourceLine" id="cb23-9" data-line-number="9">    <span class="dt">log =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(<span class="dt">x =</span> <span class="ot">NULL</span>, <span class="dt">y =</span> <span class="ot">NULL</span>), <span class="dt">mapping =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(<span class="dt">x =</span> <span class="st">"X"</span>, <span class="dt">y =</span> <span class="st">"Y"</span>, <span class="dt">colour =</span> <span class="st">"ColorBy"</span>),</a>
<a class="sourceLine" id="cb23-10" data-line-number="10">    <span class="dt">direction =</span> <span class="st">"xy"</span>, <span class="dt">brushId =</span> <span class="st">"redDimPlot1_Brush"</span>, <span class="dt">outputId =</span> <span class="st">"redDimPlot1"</span>)</a>
<a class="sourceLine" id="cb23-11" data-line-number="11">redDimArgs[[<span class="st">"ZoomData"</span>]] [[<span class="dv">2</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dt">xmin =</span> <span class="fl">3.4</span>, <span class="dt">xmax =</span> <span class="fl">27.3</span>, <span class="dt">ymin =</span> <span class="fl">0.8</span>, <span class="dt">ymax =</span> <span class="fl">30.0</span>)</a>
<a class="sourceLine" id="cb23-12" data-line-number="12">initialPanels &lt;-<span class="st"> </span><span class="kw">DataFrame</span>(</a>
<a class="sourceLine" id="cb23-13" data-line-number="13">    <span class="dt">Name=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"Reduced dimension plot 1"</span>, <span class="st">"Reduced dimension plot 2"</span>),</a>
<a class="sourceLine" id="cb23-14" data-line-number="14">    <span class="dt">Width=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rep">rep</a></span>(6L, 2L),</a>
<a class="sourceLine" id="cb23-15" data-line-number="15">    <span class="dt">Height=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rep">rep</a></span>(<span class="dv">400</span>, 2L)</a>
<a class="sourceLine" id="cb23-16" data-line-number="16">)</a>
<a class="sourceLine" id="cb23-17" data-line-number="17"><span class="cf">if</span> (<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/interactive">interactive</a></span>()) {</a>
<a class="sourceLine" id="cb23-18" data-line-number="18">    <span class="kw"><a href="https://www.rdocumentation.org/packages/iSEE/topics/iSEE">iSEE</a></span>(sce, <span class="dt">redDimArgs =</span> redDimArgs, <span class="dt">initialPanels =</span> initialPanels)</a>
<a class="sourceLine" id="cb23-19" data-line-number="19">}</a></code></pre></div>
<p><img src="img/Zoom.png"></p>
</div>
<div id="filter-table" class="section level3">
<h3 class="hasAnchor">
<a href="#filter-table" class="anchor"></a>Filtering tables</h3>
<p>Tables may also be preconfigured to immediately display a subset of rows according to their filters or select a row different from the first one when the app is launched.</p>
<p>For simplicity, we demonstrate this feature using a subset of 3 column metadata–namely “Cluster”, “total_counts”, and “pct_counts_MT”–out of the set of 48 column metadata computed by the <a href="#prepare-scrnaseq-data">workflow above</a>.</p>
<p>In the example below, we preconfigure a <em>Column statistics table</em> to display only cells with cluster labels “3” and “5”, and a library size between 5,000 and 15,844 (the maximum library size in the dataset). In addition, we also preselect the cell named “Cell684”.</p>
<blockquote>
<p><i class="fa fa-wrench fa-2x"></i> <strong>Action:</strong> Close the app and run the following block of code to launch the example application described above.</p>
</blockquote>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" data-line-number="1">sce1 &lt;-<span class="st"> </span>sce</a>
<a class="sourceLine" id="cb24-2" data-line-number="2"><span class="kw">colData</span>(sce1) &lt;-<span class="st"> </span><span class="kw">colData</span>(sce1)[, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"Cluster"</span>, <span class="st">"total_counts"</span>, <span class="st">"pct_counts_MT"</span>)]</a>
<a class="sourceLine" id="cb24-3" data-line-number="3">colStatArgs &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/iSEE/topics/colStatTableDefaults">colStatTableDefaults</a></span>(sce1, <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb24-4" data-line-number="4">colStatArgs[[<span class="st">"SearchColumns"</span>]][[<span class="dv">1</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"[</span><span class="ch">\"</span><span class="st">3</span><span class="ch">\"</span><span class="st">, </span><span class="ch">\"</span><span class="st">5</span><span class="ch">\"</span><span class="st">]"</span>, <span class="st">"5000 ... 15844"</span>, <span class="st">""</span>)</a>
<a class="sourceLine" id="cb24-5" data-line-number="5">colStatArgs[[<span class="st">"Selected"</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/match">match</a></span>(<span class="st">"Cell684"</span>, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(sce1))</a>
<a class="sourceLine" id="cb24-6" data-line-number="6">initialPanels &lt;-<span class="st"> </span><span class="kw">DataFrame</span>(</a>
<a class="sourceLine" id="cb24-7" data-line-number="7">    <span class="dt">Name=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"Column statistics table 1"</span>),</a>
<a class="sourceLine" id="cb24-8" data-line-number="8">    <span class="dt">Width=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(12L),</a>
<a class="sourceLine" id="cb24-9" data-line-number="9">    <span class="dt">Height=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">600</span>)</a>
<a class="sourceLine" id="cb24-10" data-line-number="10">)</a>
<a class="sourceLine" id="cb24-11" data-line-number="11"><span class="cf">if</span> (<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/interactive">interactive</a></span>()) {</a>
<a class="sourceLine" id="cb24-12" data-line-number="12">    <span class="kw"><a href="https://www.rdocumentation.org/packages/iSEE/topics/iSEE">iSEE</a></span>(sce1, <span class="dt">colStatArgs =</span> colStatArgs, <span class="dt">initialPanels =</span> initialPanels)</a>
<a class="sourceLine" id="cb24-13" data-line-number="13">}</a></code></pre></div>
<p><img src="img/FilteredTable.png"></p>
<p>Notably the preselected cell can be used to immediately establish a relationship with another panel and highlighted the selection in a different panel <a href="#transmit-point-selections">as demonstrated earlier</a>.</p>
<blockquote>
<p><i class="fa fa-wrench fa-2x"></i> <strong>Action:</strong> Close the app and run the following block of code to launch an application that includes a <em>Reduced dimension plot</em> panel dynamically higlighting the cell selected in the <em>Column statistics table</em>. Select different rows of the cell to see the <em>Reduced dimension plot</em> panel update accordingly.</p>
</blockquote>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" data-line-number="1">redDimArgs &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/iSEE/topics/redDimPlotDefaults">redDimPlotDefaults</a></span>(sce1, <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb25-2" data-line-number="2">redDimArgs[[<span class="st">"Type"</span>]] &lt;-<span class="st"> "TSNE"</span></a>
<a class="sourceLine" id="cb25-3" data-line-number="3">redDimArgs<span class="op">$</span>VisualBoxOpen &lt;-<span class="st"> </span><span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb25-4" data-line-number="4">redDimArgs<span class="op">$</span>ColorBy &lt;-<span class="st"> "Sample name"</span></a>
<a class="sourceLine" id="cb25-5" data-line-number="5">redDimArgs<span class="op">$</span>ColorByColTable &lt;-<span class="st"> "Column statistics table 1"</span></a>
<a class="sourceLine" id="cb25-6" data-line-number="6">initialPanels &lt;-<span class="st"> </span><span class="kw">DataFrame</span>(</a>
<a class="sourceLine" id="cb25-7" data-line-number="7">    <span class="dt">Name=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"Column statistics table 1"</span>, <span class="st">"Reduced dimension plot 1"</span>),</a>
<a class="sourceLine" id="cb25-8" data-line-number="8">    <span class="dt">Width=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(6L, 6L),</a>
<a class="sourceLine" id="cb25-9" data-line-number="9">    <span class="dt">Height=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">600</span>, <span class="dv">400</span>)</a>
<a class="sourceLine" id="cb25-10" data-line-number="10">)</a>
<a class="sourceLine" id="cb25-11" data-line-number="11"><span class="cf">if</span> (<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/interactive">interactive</a></span>()) {</a>
<a class="sourceLine" id="cb25-12" data-line-number="12">    <span class="kw"><a href="https://www.rdocumentation.org/packages/iSEE/topics/iSEE">iSEE</a></span>(sce1, <span class="dt">redDimArgs =</span> redDimArgs, <span class="dt">colStatArgs =</span> colStatArgs,</a>
<a class="sourceLine" id="cb25-13" data-line-number="13">        <span class="dt">initialPanels =</span> initialPanels)</a>
<a class="sourceLine" id="cb25-14" data-line-number="14">}</a></code></pre></div>
<p><img src="img/LinkTableToPlot.png"></p>
</div>
</div>
<div id="save-and-restore-user-interface" class="section level2">
<h2 class="hasAnchor">
<a href="#save-and-restore-user-interface" class="anchor"></a>Saving and restoring the state of the application</h2>
<p>In the previous sections we have seen how we can specify the initial state of each panel by modifying the columns of the corresponding <code>*Args</code> argument. Often, the initial exploration of a data set, including reorganization and configuration of panels is done interactively. Once a satisfactory panel setup has been achieved, the corresponding settings can be exported and used to restore the state of the app at startup.</p>
<blockquote>
<p><i class="fa fa-wrench fa-2x"></i> <strong>Action:</strong> Click on the wrench icon (<i class="fa fa-wrench"></i>) in the top-right corner, and select ‘Display panel settings’. Copy all the code shown in the pop-up window. Close the app and paste this code in your R session. Then launch a new instance using the following code. Note how the app starts in the same configuration as it was before closing. Of course, it is still possible to continue exploring the data interactively - we have only changed the starting configuration.</p>
</blockquote>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" data-line-number="1"><span class="cf">if</span> (<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/interactive">interactive</a></span>()) {</a>
<a class="sourceLine" id="cb26-2" data-line-number="2">    <span class="kw"><a href="https://www.rdocumentation.org/packages/iSEE/topics/iSEE">iSEE</a></span>(sce, <span class="dt">redDimArgs =</span> redDimPlotArgs, <span class="dt">colDataArgs =</span> colDataPlotArgs,</a>
<a class="sourceLine" id="cb26-3" data-line-number="3">         <span class="dt">featAssayArgs =</span> featAssayPlotArgs, <span class="dt">rowStatArgs =</span> rowStatTableArgs,</a>
<a class="sourceLine" id="cb26-4" data-line-number="4">         <span class="dt">rowDataArgs =</span> rowDataPlotArgs, <span class="dt">sampAssayArgs =</span> sampAssayPlotArgs,</a>
<a class="sourceLine" id="cb26-5" data-line-number="5">         <span class="dt">colStatArgs =</span> colStatTableArgs, <span class="dt">customDataArgs =</span> customDataPlotArgs,</a>
<a class="sourceLine" id="cb26-6" data-line-number="6">         <span class="dt">customStatArgs =</span> customStatTableArgs, <span class="dt">heatMapArgs =</span> heatMapPlotArgs,</a>
<a class="sourceLine" id="cb26-7" data-line-number="7">         <span class="dt">initialPanels =</span> initialPanels)</a>
<a class="sourceLine" id="cb26-8" data-line-number="8">}</a></code></pre></div>
</div>
<div id="custom-panels" class="section level2">
<h2 class="hasAnchor">
<a href="#custom-panels" class="anchor"></a>Custom panels</h2>
<p>In addition to the standard panel types, <code>R Biocpkg("iSEE")</code> allows the user to create <em>custom</em> panels, displaying any type of plot or table. The custom panels can receive row and column selections from the standard panels. In order to create a custom panel, we need to define a function that generates the desired output. This function must receive a <code>SummarizedExperiment</code> object as it first input. In addition, if only one selection is to be transmitted (i.e., if <code><a href="https://www.rdocumentation.org/packages/iSEE/topics/iSEE">iSEE(..., customSendAll=FALSE)</a></code>) it should take a vector of row names as the second argument, and a vector of column names as the third argument. If <code><a href="https://www.rdocumentation.org/packages/iSEE/topics/iSEE">iSEE(..., customSendAll=TRUE)</a></code>, all selections (active and saved) are transmitted. We will first show how to create a custom panel that performs dimension reduction (here, PCA) on a subset of the rows and columns, selected in and transmitted from other panels in the interface.</p>
<blockquote>
<p><i class="fa fa-wrench fa-2x"></i> <strong>Action:</strong> Copy the code below in your R session in order to define the custom function.</p>
</blockquote>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(scater)</a>
<a class="sourceLine" id="cb27-2" data-line-number="2">CUSTOM_PCA &lt;-<span class="st"> </span><span class="cf">function</span>(se, rows, columns,</a>
<a class="sourceLine" id="cb27-3" data-line-number="3">                       <span class="dt">colour_by =</span> <span class="ot">NULL</span>, <span class="dt">scale_features =</span> <span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb27-4" data-line-number="4">) {</a>
<a class="sourceLine" id="cb27-5" data-line-number="5">    <span class="cf">if</span> (<span class="op">!</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/NULL">is.null</a></span>(columns)) {</a>
<a class="sourceLine" id="cb27-6" data-line-number="6">        kept &lt;-<span class="st"> </span>se[, columns]</a>
<a class="sourceLine" id="cb27-7" data-line-number="7">    } <span class="cf">else</span> {</a>
<a class="sourceLine" id="cb27-8" data-line-number="8">        <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/function">return</a></span>(</a>
<a class="sourceLine" id="cb27-9" data-line-number="9">            <span class="kw">ggplot</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme_void</span>() <span class="op">+</span><span class="st"> </span><span class="kw">geom_text</span>(</a>
<a class="sourceLine" id="cb27-10" data-line-number="10">                <span class="kw">aes</span>(x, y, <span class="dt">label =</span> label),</a>
<a class="sourceLine" id="cb27-11" data-line-number="11">                <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(<span class="dt">x =</span> <span class="dv">0</span>, <span class="dt">y =</span> <span class="dv">0</span>, <span class="dt">label =</span> <span class="st">"No column data selected."</span>),</a>
<a class="sourceLine" id="cb27-12" data-line-number="12">                <span class="dt">size =</span> <span class="dv">5</span>)</a>
<a class="sourceLine" id="cb27-13" data-line-number="13">            )</a>
<a class="sourceLine" id="cb27-14" data-line-number="14">    }</a>
<a class="sourceLine" id="cb27-15" data-line-number="15"></a>
<a class="sourceLine" id="cb27-16" data-line-number="16">    scale_features &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/logical">as.logical</a></span>(scale_features)</a>
<a class="sourceLine" id="cb27-17" data-line-number="17">    kept &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/BiocSingular/topics/runPCA">runPCA</a></span>(kept, <span class="dt">feature_set =</span> rows, <span class="dt">scale_features =</span> scale_features)</a>
<a class="sourceLine" id="cb27-18" data-line-number="18">    <span class="kw"><a href="https://www.rdocumentation.org/packages/scater/topics/plot_reddim">plotPCA</a></span>(kept, <span class="dt">colour_by =</span> colour_by)</a>
<a class="sourceLine" id="cb27-19" data-line-number="19">}</a></code></pre></div>
<blockquote>
<p><i class="fa fa-wrench fa-2x"></i> <strong>Action:</strong> Next, launch an iSEE session with the custom PCA plot panel, as well as a Reduced dimension plot and a Row data plot, from which the column and row selections for the custom PCA are obtained. Select subsets of the points in each of these two panels and note how the PCA plot in the custom panel is updated. Then open the Data parameters box in the custom panel, and select another annotation (or a feature name) to color the points by. Note how iSEE detects that the input arguments were changed, and prompts you to update the plot.</p>
</blockquote>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" data-line-number="1">cdp &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/iSEE/topics/customDataPlotDefaults">customDataPlotDefaults</a></span>(sce, <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb28-2" data-line-number="2">cdp<span class="op">$</span>Function &lt;-<span class="st"> "CUSTOM_PCA"</span></a>
<a class="sourceLine" id="cb28-3" data-line-number="3">cdp<span class="op">$</span>Arguments &lt;-<span class="st"> "colour_by Cluster</span><span class="ch">\n</span><span class="st">scale_features FALSE"</span></a>
<a class="sourceLine" id="cb28-4" data-line-number="4">cdp<span class="op">$</span>ColumnSource &lt;-<span class="st"> "Reduced dimension plot 1"</span></a>
<a class="sourceLine" id="cb28-5" data-line-number="5">cdp<span class="op">$</span>RowSource &lt;-<span class="st"> "Row data plot 1"</span></a>
<a class="sourceLine" id="cb28-6" data-line-number="6"></a>
<a class="sourceLine" id="cb28-7" data-line-number="7">rdp &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/iSEE/topics/rowDataPlotDefaults">rowDataPlotDefaults</a></span>(sce, <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb28-8" data-line-number="8">rdp<span class="op">$</span>YAxis &lt;-<span class="st"> "pct_dropout_by_counts"</span></a>
<a class="sourceLine" id="cb28-9" data-line-number="9">rdp<span class="op">$</span>XAxis &lt;-<span class="st"> "Row data"</span></a>
<a class="sourceLine" id="cb28-10" data-line-number="10">rdp<span class="op">$</span>XAxisRowData &lt;-<span class="st"> "log10_mean_counts"</span></a>
<a class="sourceLine" id="cb28-11" data-line-number="11"></a>
<a class="sourceLine" id="cb28-12" data-line-number="12"><span class="cf">if</span> (<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/interactive">interactive</a></span>()) {</a>
<a class="sourceLine" id="cb28-13" data-line-number="13">  <span class="kw"><a href="https://www.rdocumentation.org/packages/iSEE/topics/iSEE">iSEE</a></span>(sce, <span class="dt">customDataArgs =</span> cdp,</a>
<a class="sourceLine" id="cb28-14" data-line-number="14">       <span class="dt">initialPanels =</span> <span class="kw">DataFrame</span>(</a>
<a class="sourceLine" id="cb28-15" data-line-number="15">         <span class="dt">Name =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(</a>
<a class="sourceLine" id="cb28-16" data-line-number="16">           <span class="st">"Reduced dimension plot 1"</span>, <span class="st">"Row data plot 1"</span>,</a>
<a class="sourceLine" id="cb28-17" data-line-number="17">           <span class="st">"Custom data plot 1"</span>),</a>
<a class="sourceLine" id="cb28-18" data-line-number="18">         <span class="dt">Width =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">4</span>)),</a>
<a class="sourceLine" id="cb28-19" data-line-number="19">       <span class="dt">customDataFun =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(<span class="dt">CUSTOM_PCA =</span> CUSTOM_PCA)</a>
<a class="sourceLine" id="cb28-20" data-line-number="20">  )</a>
<a class="sourceLine" id="cb28-21" data-line-number="21">}</a></code></pre></div>
<p>More examples of custom panels can be found at <a href="https://github.com/kevinrue/iSEE_custom" class="uri">https://github.com/kevinrue/iSEE_custom</a> and in the <a href="https://bioconductor.org/packages/release/bioc/vignettes/iSEE/inst/doc/custom.html">Custom panels vignette</a>.</p>
<!-- Here we could mention the custom coverage plot, and link to bulk RNA-seq data -->
<!-- Multiple selections -> DE -->
<!-- If time is enough: Kev's comparison of reduced dimensionality could be also nice - just to show, maybe, and then link to the iSEE_custom repo -->
</div>
<div id="reproducibility" class="section level2">
<h2 class="hasAnchor">
<a href="#reproducibility" class="anchor"></a>Reproducibility</h2>
<p>The fact that data exploration is done interactively is no reason to forego reproducibility! To this end, <em><a href="https://bioconductor.org/packages/3.10/iSEE">iSEE</a></em> lets you export the exact R code used to create each of the currently visible plots.</p>
<blockquote>
<p><i class="fa fa-wrench fa-2x"></i> <strong>Action:</strong> Close the application and run the following block of code to set up 3 panels, all displaying the t-SNE representation.</p>
</blockquote>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb29-1" data-line-number="1">redDimArgs &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/iSEE/topics/redDimPlotDefaults">redDimPlotDefaults</a></span>(sce, <span class="dv">3</span>)</a>
<a class="sourceLine" id="cb29-2" data-line-number="2">redDimArgs[[<span class="st">"Type"</span>]] &lt;-<span class="st"> "TSNE"</span></a>
<a class="sourceLine" id="cb29-3" data-line-number="3">redDimArgs[<span class="dv">1</span>, <span class="st">"ColorBy"</span>] &lt;-<span class="st"> "Column data"</span></a>
<a class="sourceLine" id="cb29-4" data-line-number="4">redDimArgs[<span class="dv">1</span>, <span class="st">"ColorByColData"</span>] &lt;-<span class="st"> "Cluster"</span></a>
<a class="sourceLine" id="cb29-5" data-line-number="5">redDimArgs[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>, <span class="st">"ColorBy"</span>] &lt;-<span class="st"> "Feature name"</span></a>
<a class="sourceLine" id="cb29-6" data-line-number="6">redDimArgs[<span class="dv">2</span>, <span class="st">"ColorByFeatName"</span>] &lt;-<span class="st"> "CD79A"</span></a>
<a class="sourceLine" id="cb29-7" data-line-number="7">redDimArgs[<span class="dv">3</span>, <span class="st">"ColorByFeatName"</span>] &lt;-<span class="st"> "CD74"</span></a>
<a class="sourceLine" id="cb29-8" data-line-number="8">initialPanels &lt;-<span class="st"> </span><span class="kw">DataFrame</span>(</a>
<a class="sourceLine" id="cb29-9" data-line-number="9">    <span class="dt">Name=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste</a></span>(<span class="st">"Reduced dimension plot"</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>),</a>
<a class="sourceLine" id="cb29-10" data-line-number="10">    <span class="dt">Width=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rep">rep</a></span>(4L, 3L),</a>
<a class="sourceLine" id="cb29-11" data-line-number="11">    <span class="dt">Height=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rep">rep</a></span>(<span class="dv">400</span>, 3L)</a>
<a class="sourceLine" id="cb29-12" data-line-number="12">)</a>
<a class="sourceLine" id="cb29-13" data-line-number="13"><span class="cf">if</span> (<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/interactive">interactive</a></span>()) {</a>
<a class="sourceLine" id="cb29-14" data-line-number="14">    <span class="kw"><a href="https://www.rdocumentation.org/packages/iSEE/topics/iSEE">iSEE</a></span>(sce, <span class="dt">redDimArgs =</span> redDimArgs, <span class="dt">initialPanels =</span> initialPanels)</a>
<a class="sourceLine" id="cb29-15" data-line-number="15">}</a></code></pre></div>
<blockquote>
<p>Then click on the wrench icon (<i class="fa fa-wrench"></i>) in the top-right corner, and select ‘Extract the R code’. Copy the preamble and the code for the first <em>Reduced dimension plot</em> from the pop-up window. Close the app and paste the code in your R session. Note how this recreates the first reduced dimension plot from the app.</p>
</blockquote>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb30-1" data-line-number="1">se &lt;-<span class="st"> </span>sce</a>
<a class="sourceLine" id="cb30-2" data-line-number="2"><span class="kw">colData</span>(se)[,<span class="st">"sizeFactors(se)"</span>] &lt;-<span class="st"> </span><span class="kw">sizeFactors</span>(se)</a>
<a class="sourceLine" id="cb30-3" data-line-number="3">colormap &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/iSEE/topics/ExperimentColorMap-class">ExperimentColorMap</a></span>()</a>
<a class="sourceLine" id="cb30-4" data-line-number="4">colormap &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/iSEE/topics/synchronizeAssays">synchronizeAssays</a></span>(colormap, se)</a>
<a class="sourceLine" id="cb30-5" data-line-number="5">all_coordinates &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>()</a>
<a class="sourceLine" id="cb30-6" data-line-number="6">custom_data_fun &lt;-<span class="st"> </span><span class="ot">NULL</span></a>
<a class="sourceLine" id="cb30-7" data-line-number="7">custom_stat_fun &lt;-<span class="st"> </span><span class="ot">NULL</span></a>
<a class="sourceLine" id="cb30-8" data-line-number="8"></a>
<a class="sourceLine" id="cb30-9" data-line-number="9">################################################################################</a>
<a class="sourceLine" id="cb30-10" data-line-number="10">## Reduced dimension plot 1</a>
<a class="sourceLine" id="cb30-11" data-line-number="11">################################################################################</a>
<a class="sourceLine" id="cb30-12" data-line-number="12"></a>
<a class="sourceLine" id="cb30-13" data-line-number="13">red.dim &lt;-<span class="st"> </span><span class="kw">reducedDim</span>(se, <span class="dv">2</span>);</a>
<a class="sourceLine" id="cb30-14" data-line-number="14">plot.data &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(<span class="dt">X =</span> red.dim[, <span class="dv">1</span>], <span class="dt">Y =</span> red.dim[, <span class="dv">2</span>], <span class="dt">row.names=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(se));</a>
<a class="sourceLine" id="cb30-15" data-line-number="15">plot.data<span class="op">$</span>ColorBy &lt;-<span class="st"> </span><span class="kw">colData</span>(se)[,<span class="st">"Cluster"</span>];</a>
<a class="sourceLine" id="cb30-16" data-line-number="16">plot.data &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/subset">subset</a></span>(plot.data, <span class="op">!</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/NA">is.na</a></span>(X) <span class="op">&amp;</span><span class="st"> </span><span class="op">!</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/NA">is.na</a></span>(Y));</a>
<a class="sourceLine" id="cb30-17" data-line-number="17"></a>
<a class="sourceLine" id="cb30-18" data-line-number="18"><span class="co"># Saving data for transmission</span></a>
<a class="sourceLine" id="cb30-19" data-line-number="19">all_coordinates[[<span class="st">'redDimPlot1'</span>]] &lt;-<span class="st"> </span>plot.data</a>
<a class="sourceLine" id="cb30-20" data-line-number="20"></a>
<a class="sourceLine" id="cb30-21" data-line-number="21"><span class="co"># Creating the plot</span></a>
<a class="sourceLine" id="cb30-22" data-line-number="22"><span class="kw">ggplot</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb30-23" data-line-number="23"><span class="st">    </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">x =</span> X, <span class="dt">y =</span> Y, <span class="dt">color =</span> ColorBy), <span class="dt">alpha =</span> <span class="dv">1</span>, plot.data, <span class="dt">size=</span><span class="dv">1</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb30-24" data-line-number="24"><span class="st">    </span><span class="kw">labs</span>(<span class="dt">x =</span> <span class="st">"Dimension 1"</span>, <span class="dt">y =</span> <span class="st">"Dimension 2"</span>, <span class="dt">color =</span> <span class="st">"Cluster"</span>, <span class="dt">title =</span> <span class="st">"(2) TSNE"</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb30-25" data-line-number="25"><span class="st">    </span><span class="kw">coord_cartesian</span>(<span class="dt">xlim =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/range">range</a></span>(plot.data<span class="op">$</span>X, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>),</a>
<a class="sourceLine" id="cb30-26" data-line-number="26">        <span class="dt">ylim =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/range">range</a></span>(plot.data<span class="op">$</span>Y, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>), <span class="dt">expand =</span> <span class="ot">TRUE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb30-27" data-line-number="27"><span class="st">    </span><span class="kw">scale_color_manual</span>(<span class="dt">values=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/iSEE/topics/ExperimentColorMap-class">colDataColorMap</a></span>(colormap, <span class="st">"Cluster"</span>, <span class="dt">discrete=</span><span class="ot">TRUE</span>)(<span class="dv">12</span>), <span class="dt">na.value=</span><span class="st">'grey50'</span>, <span class="dt">drop=</span><span class="ot">FALSE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb30-28" data-line-number="28"><span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/iSEE/topics/ExperimentColorMap-class">colDataColorMap</a></span>(colormap, <span class="st">"Cluster"</span>, <span class="dt">discrete=</span><span class="ot">TRUE</span>)(<span class="dv">12</span>), <span class="dt">na.value=</span><span class="st">'grey50'</span>, <span class="dt">drop=</span><span class="ot">FALSE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb30-29" data-line-number="29"><span class="st">    </span><span class="kw">theme_bw</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb30-30" data-line-number="30"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">'bottom'</span>, <span class="dt">legend.box =</span> <span class="st">'vertical'</span>, <span class="dt">legend.text=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">9</span>), <span class="dt">legend.title=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">11</span>),</a>
<a class="sourceLine" id="cb30-31" data-line-number="31">            <span class="dt">axis.text=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">10</span>), <span class="dt">axis.title=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">12</span>), <span class="dt">title=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">12</span>))</a></code></pre></div>
<p><img src="iSEE-lab_files/figure-html/unnamed-chunk-30-1.png" width="700"></p>
</div>
<div id="sharing-tours" class="section level2">
<h2 class="hasAnchor">
<a href="#sharing-tours" class="anchor"></a>Sharing and communication</h2>
<p>One important aspect of visualization is the ability to share your insights with others. A powerful way of easily getting people unfamiliar with your data up to speed is to provide a walk-through of the interface and the different types of plots that are displayed. With <em><a href="https://bioconductor.org/packages/3.10/iSEE">iSEE</a></em>, this can be achieved using <em>tours</em>. To configure a tour, you need to create a text file with two columns; named <code>element</code> and <code>intro</code>, with the former describing the UI element to highlight in each step, and the latter providing the descriptive text that will be displayed.</p>
<blockquote>
<p><i class="fa fa-wrench fa-2x"></i> <strong>Action:</strong> Run the code below to generate a data frame configuring a tour for the app generated in the previous step. Then launch a new instance, providing the data frame to the <code>tour</code> argument.</p>
</blockquote>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb31-1" data-line-number="1">tour &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(<span class="dt">element =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"#Welcome"</span>, <span class="st">"#redDimPlot1"</span>, <span class="st">"#redDimPlot2"</span>, <span class="st">"#Conclusion"</span>), </a>
<a class="sourceLine" id="cb31-2" data-line-number="2">                       <span class="dt">intro =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"Welcome to this tour!"</span>, <span class="st">"This is the first reduced dimension plot"</span>, <span class="st">"And here is the second one"</span>, <span class="st">"Thank you for taking this tour!"</span>),</a>
<a class="sourceLine" id="cb31-3" data-line-number="3">                       <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb31-4" data-line-number="4"><span class="cf">if</span> (<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/interactive">interactive</a></span>()) {</a>
<a class="sourceLine" id="cb31-5" data-line-number="5">    <span class="kw"><a href="https://www.rdocumentation.org/packages/iSEE/topics/iSEE">iSEE</a></span>(sce, <span class="dt">redDimArgs =</span> redDimArgs, <span class="dt">initialPanels =</span> initialPanels, <span class="dt">tour =</span> tour)</a>
<a class="sourceLine" id="cb31-6" data-line-number="6">}</a></code></pre></div>
<p>To see an example of a more complex tour in action, visit <a href="https://marionilab.cruk.cam.ac.uk/iSEE_pbmc4k/" class="uri">https://marionilab.cruk.cam.ac.uk/iSEE_pbmc4k/</a>. The code used to create this app and the associated tour is available from <a href="https://github.com/LTLA/iSEE2018/tree/master/tours/pbmc4k" class="uri">https://github.com/LTLA/iSEE2018/tree/master/tours/pbmc4k</a>.</p>
<!-- If there is enough time: draft a mini tour live (challenging: anchors might be tricky - rewarding: once it works, it is surely a OOOH thing) -->
</div>
</div>
<div id="more-resources" class="section level1">
<h1 class="hasAnchor">
<a href="#more-resources" class="anchor"></a>Additional resources</h1>
<ul>
<li>Bioconductor landing page: <a href="https://bioconductor.org/packages/iSEE/" class="uri">https://bioconductor.org/packages/iSEE/</a>
</li>
<li>Publication (F1000Research, 2018): <a href="https://f1000research.com/articles/7-741/v1" class="uri">https://f1000research.com/articles/7-741/v1</a>
</li>
<li>Deployed examples: <a href="https://marionilab.cruk.cam.ac.uk/" class="uri">https://marionilab.cruk.cam.ac.uk/</a>, code at <a href="https://github.com/ltla/iSEE2018" class="uri">https://github.com/ltla/iSEE2018</a>
<ul>
<li>Further deployments in the <a href="https://github.com/federicomarini/iSEE_instances" class="uri">https://github.com/federicomarini/iSEE_instances</a> repo</li>
<li>
<em><a href="https://bioconductor.org/packages/3.10/iSEE">iSEE</a></em> in production: <a href="http://www.teichlab.org/singlecell-treg" class="uri">http://www.teichlab.org/singlecell-treg</a>
</li>
</ul>
</li>
<li>Custom panel examples: <a href="https://github.com/kevinrue/iSEE_custom" class="uri">https://github.com/kevinrue/iSEE_custom</a>
</li>
<li>Development version (bug reports etc): <a href="https://github.com/csoneson/iSEE" class="uri">https://github.com/csoneson/iSEE</a>
</li>
</ul>
</div>
<div id="session-info" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb32-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/sessionInfo">sessionInfo</a></span>()</a>
<a class="sourceLine" id="cb32-2" data-line-number="2"><span class="co">#&gt; R Under development (unstable) (2019-05-13 r76492)</span></a>
<a class="sourceLine" id="cb32-3" data-line-number="3"><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></a>
<a class="sourceLine" id="cb32-4" data-line-number="4"><span class="co">#&gt; Running under: Ubuntu 14.04.5 LTS</span></a>
<a class="sourceLine" id="cb32-5" data-line-number="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb32-6" data-line-number="6"><span class="co">#&gt; Matrix products: default</span></a>
<a class="sourceLine" id="cb32-7" data-line-number="7"><span class="co">#&gt; BLAS:   /home/travis/R-bin/lib/R/lib/libRblas.so</span></a>
<a class="sourceLine" id="cb32-8" data-line-number="8"><span class="co">#&gt; LAPACK: /home/travis/R-bin/lib/R/lib/libRlapack.so</span></a>
<a class="sourceLine" id="cb32-9" data-line-number="9"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb32-10" data-line-number="10"><span class="co">#&gt; locale:</span></a>
<a class="sourceLine" id="cb32-11" data-line-number="11"><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></a>
<a class="sourceLine" id="cb32-12" data-line-number="12"><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></a>
<a class="sourceLine" id="cb32-13" data-line-number="13"><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></a>
<a class="sourceLine" id="cb32-14" data-line-number="14"><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></a>
<a class="sourceLine" id="cb32-15" data-line-number="15"><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></a>
<a class="sourceLine" id="cb32-16" data-line-number="16"><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></a>
<a class="sourceLine" id="cb32-17" data-line-number="17"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb32-18" data-line-number="18"><span class="co">#&gt; attached base packages:</span></a>
<a class="sourceLine" id="cb32-19" data-line-number="19"><span class="co">#&gt; [1] parallel  stats4    stats     graphics  grDevices utils     datasets </span></a>
<a class="sourceLine" id="cb32-20" data-line-number="20"><span class="co">#&gt; [8] methods   base     </span></a>
<a class="sourceLine" id="cb32-21" data-line-number="21"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb32-22" data-line-number="22"><span class="co">#&gt; other attached packages:</span></a>
<a class="sourceLine" id="cb32-23" data-line-number="23"><span class="co">#&gt;  [1] iSEE_1.5.1                  BiocSingular_1.1.0         </span></a>
<a class="sourceLine" id="cb32-24" data-line-number="24"><span class="co">#&gt;  [3] scran_1.13.1                scater_1.13.0              </span></a>
<a class="sourceLine" id="cb32-25" data-line-number="25"><span class="co">#&gt;  [5] ggplot2_3.1.1               TENxPBMCData_1.3.0         </span></a>
<a class="sourceLine" id="cb32-26" data-line-number="26"><span class="co">#&gt;  [7] HDF5Array_1.13.1            rhdf5_2.29.0               </span></a>
<a class="sourceLine" id="cb32-27" data-line-number="27"><span class="co">#&gt;  [9] SingleCellExperiment_1.7.0  SummarizedExperiment_1.15.0</span></a>
<a class="sourceLine" id="cb32-28" data-line-number="28"><span class="co">#&gt; [11] DelayedArray_0.11.0         BiocParallel_1.19.0        </span></a>
<a class="sourceLine" id="cb32-29" data-line-number="29"><span class="co">#&gt; [13] matrixStats_0.54.0          Biobase_2.45.0             </span></a>
<a class="sourceLine" id="cb32-30" data-line-number="30"><span class="co">#&gt; [15] GenomicRanges_1.37.1        GenomeInfoDb_1.21.0        </span></a>
<a class="sourceLine" id="cb32-31" data-line-number="31"><span class="co">#&gt; [17] IRanges_2.19.2              S4Vectors_0.23.1           </span></a>
<a class="sourceLine" id="cb32-32" data-line-number="32"><span class="co">#&gt; [19] BiocGenerics_0.31.2         BiocStyle_2.13.0           </span></a>
<a class="sourceLine" id="cb32-33" data-line-number="33"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb32-34" data-line-number="34"><span class="co">#&gt; loaded via a namespace (and not attached):</span></a>
<a class="sourceLine" id="cb32-35" data-line-number="35"><span class="co">#&gt;   [1] Rtsne_0.15                    ggbeeswarm_0.6.0             </span></a>
<a class="sourceLine" id="cb32-36" data-line-number="36"><span class="co">#&gt;   [3] colorspace_1.4-1              dynamicTreeCut_1.63-1        </span></a>
<a class="sourceLine" id="cb32-37" data-line-number="37"><span class="co">#&gt;   [5] rprojroot_1.3-2               XVector_0.25.0               </span></a>
<a class="sourceLine" id="cb32-38" data-line-number="38"><span class="co">#&gt;   [7] BiocNeighbors_1.3.0           fs_1.3.1                     </span></a>
<a class="sourceLine" id="cb32-39" data-line-number="39"><span class="co">#&gt;   [9] roxygen2_6.1.1                DT_0.6                       </span></a>
<a class="sourceLine" id="cb32-40" data-line-number="40"><span class="co">#&gt;  [11] bit64_0.9-7                   interactiveDisplayBase_1.23.0</span></a>
<a class="sourceLine" id="cb32-41" data-line-number="41"><span class="co">#&gt;  [13] AnnotationDbi_1.47.0          xml2_1.2.0                   </span></a>
<a class="sourceLine" id="cb32-42" data-line-number="42"><span class="co">#&gt;  [15] splines_3.7.0                 codetools_0.2-16             </span></a>
<a class="sourceLine" id="cb32-43" data-line-number="43"><span class="co">#&gt;  [17] knitr_1.22                    jsonlite_1.6                 </span></a>
<a class="sourceLine" id="cb32-44" data-line-number="44"><span class="co">#&gt;  [19] dbplyr_1.4.0                  shinydashboard_0.7.1         </span></a>
<a class="sourceLine" id="cb32-45" data-line-number="45"><span class="co">#&gt;  [21] shiny_1.3.2                   rentrez_1.2.2                </span></a>
<a class="sourceLine" id="cb32-46" data-line-number="46"><span class="co">#&gt;  [23] BiocManager_1.30.4            compiler_3.7.0               </span></a>
<a class="sourceLine" id="cb32-47" data-line-number="47"><span class="co">#&gt;  [25] httr_1.4.0                    dqrng_0.2.0                  </span></a>
<a class="sourceLine" id="cb32-48" data-line-number="48"><span class="co">#&gt;  [27] backports_1.1.4               assertthat_0.2.1             </span></a>
<a class="sourceLine" id="cb32-49" data-line-number="49"><span class="co">#&gt;  [29] Matrix_1.2-17                 lazyeval_0.2.2               </span></a>
<a class="sourceLine" id="cb32-50" data-line-number="50"><span class="co">#&gt;  [31] limma_3.41.0                  later_0.8.0                  </span></a>
<a class="sourceLine" id="cb32-51" data-line-number="51"><span class="co">#&gt;  [33] htmltools_0.3.6               tools_3.7.0                  </span></a>
<a class="sourceLine" id="cb32-52" data-line-number="52"><span class="co">#&gt;  [35] rsvd_1.0.0                    igraph_1.2.4.1               </span></a>
<a class="sourceLine" id="cb32-53" data-line-number="53"><span class="co">#&gt;  [37] gtable_0.3.0                  glue_1.3.1                   </span></a>
<a class="sourceLine" id="cb32-54" data-line-number="54"><span class="co">#&gt;  [39] GenomeInfoDbData_1.2.1        reshape2_1.4.3               </span></a>
<a class="sourceLine" id="cb32-55" data-line-number="55"><span class="co">#&gt;  [41] dplyr_0.8.0.1                 rappdirs_0.3.1               </span></a>
<a class="sourceLine" id="cb32-56" data-line-number="56"><span class="co">#&gt;  [43] Rcpp_1.0.1                    pkgdown_1.3.0                </span></a>
<a class="sourceLine" id="cb32-57" data-line-number="57"><span class="co">#&gt;  [45] nlme_3.1-140                  ExperimentHub_1.11.1         </span></a>
<a class="sourceLine" id="cb32-58" data-line-number="58"><span class="co">#&gt;  [47] rintrojs_0.2.0                DelayedMatrixStats_1.7.0     </span></a>
<a class="sourceLine" id="cb32-59" data-line-number="59"><span class="co">#&gt;  [49] xfun_0.6                      stringr_1.4.0                </span></a>
<a class="sourceLine" id="cb32-60" data-line-number="60"><span class="co">#&gt;  [51] beachmat_2.1.0                mime_0.6                     </span></a>
<a class="sourceLine" id="cb32-61" data-line-number="61"><span class="co">#&gt;  [53] miniUI_0.1.1.1                irlba_2.3.3                  </span></a>
<a class="sourceLine" id="cb32-62" data-line-number="62"><span class="co">#&gt;  [55] XML_3.98-1.19                 statmod_1.4.30               </span></a>
<a class="sourceLine" id="cb32-63" data-line-number="63"><span class="co">#&gt;  [57] shinyAce_0.3.3                AnnotationHub_2.17.3         </span></a>
<a class="sourceLine" id="cb32-64" data-line-number="64"><span class="co">#&gt;  [59] edgeR_3.27.1                  zlibbioc_1.31.0              </span></a>
<a class="sourceLine" id="cb32-65" data-line-number="65"><span class="co">#&gt;  [61] MASS_7.3-51.4                 scales_1.0.0                 </span></a>
<a class="sourceLine" id="cb32-66" data-line-number="66"><span class="co">#&gt;  [63] colourpicker_1.0              promises_1.0.1               </span></a>
<a class="sourceLine" id="cb32-67" data-line-number="67"><span class="co">#&gt;  [65] yaml_2.2.0                    curl_3.3                     </span></a>
<a class="sourceLine" id="cb32-68" data-line-number="68"><span class="co">#&gt;  [67] memoise_1.1.0                 gridExtra_2.3                </span></a>
<a class="sourceLine" id="cb32-69" data-line-number="69"><span class="co">#&gt;  [69] stringi_1.4.3                 RSQLite_2.1.1                </span></a>
<a class="sourceLine" id="cb32-70" data-line-number="70"><span class="co">#&gt;  [71] desc_1.2.0                    rlang_0.3.4                  </span></a>
<a class="sourceLine" id="cb32-71" data-line-number="71"><span class="co">#&gt;  [73] pkgconfig_2.0.2               commonmark_1.7               </span></a>
<a class="sourceLine" id="cb32-72" data-line-number="72"><span class="co">#&gt;  [75] bitops_1.0-6                  evaluate_0.13                </span></a>
<a class="sourceLine" id="cb32-73" data-line-number="73"><span class="co">#&gt;  [77] lattice_0.20-38               purrr_0.3.2                  </span></a>
<a class="sourceLine" id="cb32-74" data-line-number="74"><span class="co">#&gt;  [79] Rhdf5lib_1.7.0                labeling_0.3                 </span></a>
<a class="sourceLine" id="cb32-75" data-line-number="75"><span class="co">#&gt;  [81] htmlwidgets_1.3               cowplot_0.9.4                </span></a>
<a class="sourceLine" id="cb32-76" data-line-number="76"><span class="co">#&gt;  [83] bit_1.1-14                    tidyselect_0.2.5             </span></a>
<a class="sourceLine" id="cb32-77" data-line-number="77"><span class="co">#&gt;  [85] plyr_1.8.4                    magrittr_1.5                 </span></a>
<a class="sourceLine" id="cb32-78" data-line-number="78"><span class="co">#&gt;  [87] bookdown_0.10                 R6_2.4.0                     </span></a>
<a class="sourceLine" id="cb32-79" data-line-number="79"><span class="co">#&gt;  [89] DBI_1.0.0                     mgcv_1.8-28                  </span></a>
<a class="sourceLine" id="cb32-80" data-line-number="80"><span class="co">#&gt;  [91] pillar_1.4.0                  withr_2.1.2                  </span></a>
<a class="sourceLine" id="cb32-81" data-line-number="81"><span class="co">#&gt;  [93] RCurl_1.95-4.12               tibble_2.1.1                 </span></a>
<a class="sourceLine" id="cb32-82" data-line-number="82"><span class="co">#&gt;  [95] crayon_1.3.4                  BiocFileCache_1.9.0          </span></a>
<a class="sourceLine" id="cb32-83" data-line-number="83"><span class="co">#&gt;  [97] rmarkdown_1.12                viridis_0.5.1                </span></a>
<a class="sourceLine" id="cb32-84" data-line-number="84"><span class="co">#&gt;  [99] locfit_1.5-9.1                grid_3.7.0                   </span></a>
<a class="sourceLine" id="cb32-85" data-line-number="85"><span class="co">#&gt; [101] blob_1.1.1                    digest_0.6.18                </span></a>
<a class="sourceLine" id="cb32-86" data-line-number="86"><span class="co">#&gt; [103] xtable_1.8-4                  httpuv_1.5.1                 </span></a>
<a class="sourceLine" id="cb32-87" data-line-number="87"><span class="co">#&gt; [105] munsell_0.5.0                 beeswarm_0.2.3               </span></a>
<a class="sourceLine" id="cb32-88" data-line-number="88"><span class="co">#&gt; [107] viridisLite_0.3.0             vipor_0.4.5                  </span></a>
<a class="sourceLine" id="cb32-89" data-line-number="89"><span class="co">#&gt; [109] shinyjs_1.0</span></a></code></pre></div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-Bremner1994-hintonplots">
<p>Bremner, Frederick J, Stephen J Gotts, and Dina L Denham. 1994. “Hinton Diagrams: Viewing Connection Strengths in Neural Networks.” <em>Behav. Res. Methods Instrum. Comput.</em> 26 (2): 215–18.</p>
</div>
<div id="ref-Hinton1991-hintondiagram">
<p>Hinton, Geoffrey E, and Tim Shallice. 1991. “Lesioning an Attractor Network: Investigations of Acquired Dyslexia.” <em>Psychological Review</em> 98 (1): 74–95.</p>
</div>
</div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p>University of Oxford, Oxford, UK<a href="#fnref1" class="footnote-back">↩</a></p></li>
<li id="fn2"><p>Friedrich Miescher Institute for Biomedical Research and SIB Swiss Institute of Bioinformatics<a href="#fnref2" class="footnote-back">↩</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#instructor-names-and-contact-information">Instructor names and contact information</a></li>
      <li>
<a href="#workshop-description">Workshop description</a><ul class="nav nav-pills nav-stacked">
<li><a href="#prerequisites">Pre-requisites</a></li>
      <li><a href="#participation">Workshop Participation</a></li>
      <li><a href="#packages">R/Bioconductor packages used</a></li>
      <li><a href="#outline">Time outline</a></li>
      </ul>
</li>
      <li>
<a href="#goals-objectives">Workshop goals and objectives</a><ul class="nav nav-pills nav-stacked">
<li><a href="#goals">Learning goals</a></li>
      <li><a href="#objectives">Learning objectives</a></li>
      </ul>
</li>
      <li><a href="#prepare-scrnaseq-data">Preparation of example scRNAseq data</a></li>
      <li>
<a href="#isee-overview">Overview of the <em>iSEE</em> package</a><ul class="nav nav-pills nav-stacked">
<li><a href="#user-interface">The user interface</a></li>
      <li><a href="#configuration-ui">Configuration of the app interface</a></li>
      <li><a href="#save-and-restore-user-interface">Saving and restoring the state of the application</a></li>
      <li><a href="#custom-panels">Custom panels</a></li>
      <li><a href="#reproducibility">Reproducibility</a></li>
      <li><a href="#sharing-tours">Sharing and communication</a></li>
      </ul>
</li>
      <li><a href="#more-resources">Additional resources</a></li>
      <li><a href="#session-info">Session info</a></li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Kevin Rue-Albrecht, Federico Marini, Charlotte Soneson, Aaron Lun.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
